--- cinelerra-7-original/quicktime/thirdparty/x264.082315/x264.c	2015-08-24 04:04:26.000000000 +0200
+++ cinelerra-7/quicktime/thirdparty/x264.082315/x264.c	2017-10-16 23:38:07.549920608 +0200
@@ -442,7 +442,7 @@
     printf( "                              - valid csps for `lavf' demuxer:\n" );
     printf( INDENT );
     size_t line_len = strlen( INDENT );
-    for( enum PixelFormat i = AV_PIX_FMT_NONE+1; i < AV_PIX_FMT_NB; i++ )
+    for( enum AVPixelFormat i = AV_PIX_FMT_NONE+1; i < AV_PIX_FMT_NB; i++ )
     {
         const char *pfname = av_get_pix_fmt_name( i );
         if( pfname )
