--- cinelerra-7-original/cinelerra/mwindow.C	2015-11-23 18:42:14.000000000 +0100
+++ cinelerra-7/cinelerra/mwindow.C	2017-10-20 13:14:09.093416129 +0200
@@ -107,7 +107,7 @@
 #include "zoombar.h"
 
 #include <string.h>
-
+#include <sys/stat.h>
 
 
 extern "C"
@@ -376,7 +376,7 @@
 	char index_path[BCTEXTLEN];
 	FILE *index_fd = 0;
 	
-	sprintf(index_path, "%s/%s", path, PLUGIN_FILE);
+	sprintf(index_path, "%s/%s", preferences->index_directory, PLUGIN_FILE);
 //printf("MWindow::init_plugin_path %d %s plugindb=%p\n", __LINE__, index_path, plugindb);
 
 
@@ -401,8 +401,10 @@
 // Create plugin server from index entry
 				PluginServer *new_plugin = new PluginServer(path);
 				result = new_plugin->read_table(index_line);
+				struct stat buffer;
+				int exists = stat(new_plugin->path, &buffer);
 //printf("%p new_plugin=%p %s", result2, new_plugin, index_line);
-				if(!result)
+				if(!result && !exists)
 				{
 					plugindb->append(new_plugin);
 				}
