diff -Naur a/meson.build b/meson.build
--- a/meson.build	2022-12-20 19:15:44.000000000 +0100
+++ b/meson.build	2022-12-26 21:41:13.671772944 +0100
@@ -3515,7 +3515,7 @@
         include_directories : includes,
         link_with : [libshared],
         install_rpath : rootpkglibdir,
-        install : (conf.get('HAVE_SYSV_COMPAT') == 1),
+        install : true,
         install_dir : rootlibexecdir)
 
 public_programs += executable(
diff -Naur a/rules.d/50-udev-default.rules.in b/rules.d/50-udev-default.rules.in
--- a/rules.d/50-udev-default.rules.in	2022-12-20 19:15:44.000000000 +0100
+++ b/rules.d/50-udev-default.rules.in	2022-12-26 21:43:05.163000639 +0100
@@ -40,8 +40,8 @@
 SUBSYSTEM=="media", GROUP="video"
 SUBSYSTEM=="cec", GROUP="video"
 
-SUBSYSTEM=="drm", KERNEL=="renderD*", GROUP="render", MODE="{{GROUP_RENDER_MODE}}"
-SUBSYSTEM=="kfd", GROUP="render", MODE="{{GROUP_RENDER_MODE}}"
+SUBSYSTEM=="drm", KERNEL=="renderD*", MODE="{{GROUP_RENDER_MODE}}"
+SUBSYSTEM=="kfd", MODE="{{GROUP_RENDER_MODE}}"
 
 SUBSYSTEM=="misc", KERNEL=="sgx_enclave", GROUP="sgx", MODE="0660"
 SUBSYSTEM=="misc", KERNEL=="sgx_vepc", GROUP="sgx", MODE="0660"
diff -Naur a/src/basic/path-util.h b/src/basic/path-util.h
--- a/src/basic/path-util.h	2022-12-20 19:15:44.000000000 +0100
+++ b/src/basic/path-util.h	2022-12-26 21:44:57.767540138 +0100
@@ -10,8 +10,8 @@
 #include "strv.h"
 #include "time-util.h"
 
-#define PATH_SPLIT_SBIN_BIN(x) x "sbin:" x "bin"
-#define PATH_SPLIT_SBIN_BIN_NULSTR(x) x "sbin\0" x "bin\0"
+#define PATH_SPLIT_SBIN_BIN(x) x "bin:" x "sbin"
+#define PATH_SPLIT_SBIN_BIN_NULSTR(x) x "bin\0" x "sbin\0"
 
 #define PATH_NORMAL_SBIN_BIN(x) x "bin"
 #define PATH_NORMAL_SBIN_BIN_NULSTR(x) x "bin\0"
diff -Naur a/src/basic/unit-def.h b/src/basic/unit-def.h
--- a/src/basic/unit-def.h	2022-12-20 19:15:44.000000000 +0100
+++ b/src/basic/unit-def.h	2022-12-26 21:46:01.426337340 +0100
@@ -176,8 +176,8 @@
 
 typedef enum SwapState {
         SWAP_DEAD,
-        SWAP_ACTIVATING,               /* /sbin/swapon is running, but the swap not yet enabled. */
-        SWAP_ACTIVATING_DONE,          /* /sbin/swapon is running, and the swap is done. */
+        SWAP_ACTIVATING,               /* /usr/sbin/swapon is running, but the swap not yet enabled. */
+        SWAP_ACTIVATING_DONE,          /* /usr/sbin/swapon is running, and the swap is done. */
         SWAP_ACTIVE,
         SWAP_DEACTIVATING,
         SWAP_DEACTIVATING_SIGTERM,
diff -Naur a/src/core/swap.c b/src/core/swap.c
--- a/src/core/swap.c	2022-12-20 19:15:44.000000000 +0100
+++ b/src/core/swap.c	2022-12-26 21:47:39.564482790 +0100
@@ -827,7 +827,7 @@
                 }
         }
 
-        r = exec_command_set(s->control_command, "/sbin/swapon", NULL);
+        r = exec_command_set(s->control_command, "/usr/sbin/swapon", NULL);
         if (r < 0)
                 goto fail;
 
@@ -865,7 +865,7 @@
         s->control_command = s->exec_command + SWAP_EXEC_DEACTIVATE;
 
         r = exec_command_set(s->control_command,
-                             "/sbin/swapoff",
+                             "/usr/sbin/swapoff",
                              s->what,
                              NULL);
         if (r < 0)
@@ -1385,7 +1385,7 @@
 
                 /* Dispatch this before we dispatch SIGCHLD, so that
                  * we always get the events from /proc/swaps before
-                 * the SIGCHLD of /sbin/swapon. */
+                 * the SIGCHLD of /usr/sbin/swapon. */
                 r = sd_event_source_set_priority(m->swap_event_source, SD_EVENT_PRIORITY_NORMAL-10);
                 if (r < 0) {
                         log_error_errno(r, "Failed to change /proc/swaps priority: %m");
diff -Naur a/src/resolve/meson.build b/src/resolve/meson.build
--- a/src/resolve/meson.build	2022-12-20 19:15:44.000000000 +0100
+++ b/src/resolve/meson.build	2022-12-26 21:50:03.088436551 +0100
@@ -174,55 +174,6 @@
 
 ############################################################
 
-tests += [
-        [files('test-resolve-tables.c'),
-         [libsystemd_resolve_core,
-          libshared],
-         [lib_openssl_or_gcrypt,
-          libm]],
-
-        [files('test-dns-packet.c'),
-         [libsystemd_resolve_core,
-          libshared],
-         [lib_openssl_or_gcrypt,
-          libm]],
-
-        [files('test-resolved-etc-hosts.c',
-               'resolved-etc-hosts.c',
-               'resolved-etc-hosts.h'),
-         [libsystemd_resolve_core,
-          libshared],
-         [lib_openssl_or_gcrypt,
-          libm]],
-
-        [files('test-resolved-packet.c'),
-         [libsystemd_resolve_core,
-          libshared],
-         [lib_openssl_or_gcrypt,
-          libm]],
-
-        [files('test-resolved-stream.c')
-         + basic_dns_sources + systemd_resolved_sources,
-         [libshared],
-         [lib_openssl_or_gcrypt,
-          libm]
-         + systemd_resolved_dependencies,
-         resolve_includes],
-
-        [files('test-dnssec.c'),
-         [libsystemd_resolve_core,
-          libshared],
-         [lib_openssl_or_gcrypt,
-          libm],
-         [], 'HAVE_OPENSSL_OR_GCRYPT'],
-
-        [files('test-dnssec-complex.c'),
-         [libsystemd_resolve_core,
-          libshared],
-         [lib_openssl_or_gcrypt,
-          libm],
-         [], '', 'manual'],
-]
 
 fuzzers += [
         [files('fuzz-dns-packet.c'),
diff -Naur a/units/modprobe@.service b/units/modprobe@.service
--- a/units/modprobe@.service	2022-12-20 19:15:44.000000000 +0100
+++ b/units/modprobe@.service	2022-12-26 21:50:34.894501916 +0100
@@ -17,4 +17,4 @@
 
 [Service]
 Type=oneshot
-ExecStart=-/sbin/modprobe -abq %I
+ExecStart=-modprobe -abq %I
