diff -Naur a/scripts/gcc-plugins/gcc-common.h b/scripts/gcc-plugins/gcc-common.h
--- a/scripts/gcc-plugins/gcc-common.h	2021-10-31 21:53:10.000000000 +0100
+++ b/scripts/gcc-plugins/gcc-common.h	2023-05-12 09:51:23.337849970 +0200
@@ -77,6 +77,8 @@
 #include "varasm.h"
 #include "stor-layout.h"
 #include "internal-fn.h"
+#include "gimple.h"
+#include "gimple-iterator.h"
 #include "gimple-expr.h"
 #include "gimple-fold.h"
 #include "context.h"
