diff --git a/scripts/init.in b/scripts/init.in
index 653b97d..e605ebf 100644
--- a/scripts/init.in
+++ b/scripts/init.in
@@ -202,7 +202,7 @@ do_mount_in_ram()
 
 }
 find_media() {
-	local PI FSIZE MSIZE LABEL
+	local LABEL
 	if [ -n "$fromiso" ]; then
 		mkdir -p /media/fromdev
 		mount -o ro ${fromdev} /media/fromdev
@@ -210,39 +210,35 @@ find_media() {
 		LOOPDEV=$(losetup -f)
 		losetup $LOOPDEV /media/fromdev${fromiso}
 	fi
-
-	for SYS in /sys/block/*
+	printf "Search the media: "
+	mkdir -p /media/cdrom
+	while [ ! -L /dev/${ISOLABEL} ]
 	do
-		[ "$verbose" == "no" ] || printf "Looking in %s for media\n" $SYS
-		[ ! -d "$SYS" ] && continue
-		DEV=/dev/$(basename $SYS)
-		[ ! -b $DEV ] && continue
-		eval $(blkid $DEV|cut -d ":" -f2)
-		if [ ${ISOLABEL} == ${LABEL} ]  2>/dev/null ; then
-		    if [ -b ${DEV}1 ]; then
-				ln -sf ${DEV}1 /dev/${LABEL}
-			else
-				ln -sf ${DEV} /dev/${LABEL}
+		for SYS in /sys/block/*
+		do
+			[ "$verbose" == "no" ] || printf "Looking in %s for media\n" $SYS
+			[ ! -d $SYS ] && continue
+			DEV="/dev/$(basename $SYS)"
+			[ ! -b $DEV ] && continue
+			eval $(blkid $DEV|cut -d ":" -f2)
+			[ "$verbose" == "no" ] || printf "label %s found\n" $LABEL
+			if [ "${ISOLABEL}" == "${LABEL}" ]; then
+				if [ -b ${DEV}1 ]; then
+					[ "$verbose" == "no" ] || printf "ln -sf ${DEV}1 /dev/${ISOLABEL}\n"
+					ln -sf ${DEV}1 /dev/${LABEL}
+				else
+					[ "$verbose" == "no" ] || printf "ln -sf ${DEV} /dev/${ISOLABEL}\n"
+					ln -sf ${DEV} /dev/${LABEL}
+				fi
+				[ "$verbose" == "no" ] || printf "mount -n $DEV /media/cdrom\n"
+				mount -n $DEV /media/cdrom
+				break
 			fi
-			mkdir  -p /media/cdrom 2>/dev/null
-			[ "$verbose" == "no" ] || printf "mount -n /dev/${LABEL} /media/cdrom\n"
-			mount -n /dev/${LABEL} /media/cdrom
-			break
-		fi
-	done
-}
-do_mount_iso() {
-	[ "$verbose" == "no" ] || printf "Looking in %s for media\n" ${LABEL}
-	while true
-	do
-		if [ -b /dev/disk/by-label/${ISOLABEL} ]; then
-			ln -sf $(readlink -f /dev/disk/by-label/${ISOLABEL}) /dev/${ISOLABEL}
-			mkdir  -p /media/cdrom
-			[ "$verbose" == "no" ] || printf "mount -n /dev/${ISOLABEL} /media/cdrom\n"
-			mount -n /dev/${ISOLABEL} /media/cdrom
-			break
-		fi
+		done
+		printf "."
+		sleep 1
 	done
+	printf "\n"
 }
 do_resize_rootfs() {
 	MEM_TOTAL="$(cat /proc/meminfo|grep MemTotal|sed "s/[^[:digit:]]*//g")"
@@ -401,15 +397,7 @@ if [ -n "$rootdelay" ]; then
 fi
 if [ -z $root ]; then
 	mkdir -p /mnt/hd
-	if [ ! -f ${SETUP_INIT_FILE} ]; then
-#		do_mount_iso
-		find_media
-		if [ ! -L /dev/${ISOLABEL} ]; then
-			[ "$verbose" == "no" ] || printf "Media not found yet, trying again in 5 sec\n"
-			sleep 5
-			find_media
-		fi
-	fi
+	find_media
 	do_resize_rootfs
 	do_mount_in_ram
 	echo "0" > /proc/sys/vm/swappiness
