diff --git a/tools/lib/init.in b/tools/lib/init.in
index 41e2d3f..26a90e0 100644
--- a/tools/lib/init.in
+++ b/tools/lib/init.in
@@ -3,6 +3,7 @@
 PATH=/bin:/usr/bin:/sbin:/usr/sbin
 export PATH
 
+
 problem()
 {
    printf "Encountered a problem!\n\nDropping you to a shell.\n\n"
@@ -114,12 +115,18 @@ do_mount_in_ram()
 }
 
 find_media() {
+   local PI
    EXPECT_LABEL="nutyxcd"
    for SYS in /sys/block/sd* /sys/block/sr* ; do
 	   [ "$verbose" == "no" ] || echo "Looking in $SYS for media"
       if [ ! -d "$SYS" ]; then  continue; fi
       DEV=/dev/${SYS##*/}
-      LABEL=`dd if=$DEV bs=1 skip=32808 count=32 2>/dev/null`
+      PI=`/sbin/blkid $DEV`
+      for param in $PI; do
+        case $param in LABEL=*)
+          LABEL=`echo ${param#LABEL=}|sed s/\"//g`;;
+	esac
+      done
       if [ ${LABEL} == ${EXPECT_LABEL} ]  2>/dev/null ; then
 	ln -s $DEV /dev/nutyx-cd
 	mkdir  -p /media/cdrom 2>/dev/null
