diff --git a/scripts/setup-nutyx.in b/scripts/setup-nutyx.in
index 384d89ede..202d34e33 100755
--- a/scripts/setup-nutyx.in
+++ b/scripts/setup-nutyx.in
@@ -61,7 +61,7 @@ _msgfmt() {
 	localedef -i ${1} -f UTF-8 ${1}.UTF-8
 	if [ -f /lib/services/setup-nutyx/setup-nutyx.locales.gz ];then
 		bsdtar -xf /lib/services/setup-nutyx/setup-nutyx.locales.gz \
-		-C $TD lib/{setup-nutyx,xdg-user-dirs,dialog}.${L}
+		-C $TD lib/{setup-nutyx,xdg-user-dirs,dialog,util-linux}.${L}
 	fi
 	if [ -f $TD/lib/setup-nutyx.${L} ]; then
 		TEXTDOMAINFILE1=$TEXTDOMAINDIR/${L}/LC_MESSAGES/setup-nutyx.mo
@@ -82,7 +82,7 @@ _msgfmt() {
 		-o $TEXTDOMAINFILE3
 	fi
 	if [ -f $TD/lib/util-linux.${L} ]; then
-		TEXTDOMAINFILE3=$TEXTDOMAINDIR/${L}/LC_MESSAGES/util-linux.mo
+		TEXTDOMAINFILE4=$TEXTDOMAINDIR/${L}/LC_MESSAGES/util-linux.mo
 		mkdir -p $TEXTDOMAINDIR/${L}/LC_MESSAGES
 		msgfmt $TD/lib/util-linux.${L} \
 		-o $TEXTDOMAINFILE4
diff --git a/scripts/wget_nutyx_list b/scripts/wget_nutyx_list
index cc2ab55c7..3abcbf9a1 100644
--- a/scripts/wget_nutyx_list
+++ b/scripts/wget_nutyx_list
@@ -90,4 +90,4 @@ fe7c0f7e439f08970e462c9d44599903  util-linux-2.36.tar.xz
 db4e6dc7977cbddcd543b240079a4899  wget-1.20.3.tar.gz
 aa1621ec7013a19abab52a8aff04fe5b  xz-5.2.5.tar.xz
 85adef240c5f370b308da8c938951a68  zlib-1.2.11.tar.xz
-2e74443f4548431205ebc45b20e1791e  nutyx-12-beta3.tar.xz
+19bc328551363a11d25ad1e132d9b094  nutyx-12-beta4.tar.xz
