diff --git a/base/linux-mainline/Pkgfile b/base/linux-mainline/Pkgfile
index 7eae40356..90cd1cb82 100644
--- a/base/linux-mainline/Pkgfile
+++ b/base/linux-mainline/Pkgfile
@@ -11,7 +11,7 @@ PKGMK_NO_STRIP="yes"
 PKGMK_GROUPS=(devel doc)
 
 name=linux-mainline
-version=5.9-rc6
+version=5.11-rc1
 
 INITRD="initrd-mainline"
 KERNEL=${name/-*/}
diff --git a/gui-extra/dillo/Pkgfile b/gui-extra/dillo/Pkgfile
index e03b9ba90..5a633f5fd 100644
--- a/gui-extra/dillo/Pkgfile
+++ b/gui-extra/dillo/Pkgfile
@@ -8,7 +8,7 @@ makedepends=(fltk openssl)
 
 name=dillo
 version=3.0.5
-release=6
+release=7
 
 source=(dillo-3.0.5-openssl-1.patch
 	http://www.dillo.org/download/dillo-$version.tar.bz2
diff --git a/gui-extra/xcfa/Pkgfile b/gui-extra/xcfa/Pkgfile
index 3413e414d..4e7ae1290 100644
--- a/gui-extra/xcfa/Pkgfile
+++ b/gui-extra/xcfa/Pkgfile
@@ -4,7 +4,7 @@ url="http://www.xcfa.tuxfamily.org/"
 packager="tnut <tnut@nutyx.org>"
 contributors="Greg"
 
-makedepends=(taglib alsa-lib curl vte3)
+makedepends=(taglib alsa-lib curl vte)
 
 name=xcfa
 version=5.0.2
diff --git a/tools/lib/init-functions b/tools/lib/init-functions
index 2a6009f8f..8e2aef4b6 100644
--- a/tools/lib/init-functions
+++ b/tools/lib/init-functions
@@ -1652,25 +1652,31 @@ do_stop_sysklogd()
 	log_info_msg "Stopping kernel log daemon..."
 	killproc $KLOGDCMD
 	evaluate_retval
-	mv /var/log/kern.log{,.old}
+	mv /var/log/kern.log \
+	/var/log/kern.log.old
 	touch /var/log/kern.log
 
 	log_info_msg "Stopping system log daemon..."
 	killproc $SYSLOGDCMD
 	evaluate_retval
-	mv /var/log/sys.log{,.old}
+	mv /var/log/sys.log \
+	/var/log/sys.log.old
 	touch /var/log/sys.log
 
-	mv /var/log/auth.log{,.old}
+	mv /var/log/auth.log \
+	/var/log/auth.log.old
 	touch /var/log/auth.log
 
-	mv /var/log/daemon.log{,.old}
+	mv /var/log/daemon.log \
+	/var/log/daemon.log.old
 	touch /var/log/daemon.log
 
-	mv /var/log/user.log{,.old}
+	mv /var/log/user.log \
+	/var/log/user.log.old
 	touch /var/log/user.log
 
-	mv /var/log/wtmp{,.old}
+	mv /var/log/wtmp \
+	/var/log/wtmp.old
 	touch /var/log/wtmp
 }
 do_status_sysklogd()
@@ -1691,11 +1697,13 @@ do_cleanup_syslogs()
 	for file in sys daemon kern mail auth user boot cron
 	do
 	if [ -f /var/log/$file.log ]; then
-		mv /var/log/$file.log{,.old}
+		mv /var/log/$file.log \
+		/var/log/$file.log.old
 		touch /var/log/$file.log
 	fi
 	done
-	mv /var/log/wtmp{,.old}
+	mv /var/log/wtmp \
+	/var/log/wtmp.old
 	touch /var/log/wtmp
 	evaluate_retval
 }
