--- a/scripts/setup-nutyx.in	2020-10-15 17:58:52.000000000 +0200
+++ b/scripts/setup-nutyx.in	2020-10-17 16:36:10.557320573 +0200
@@ -61,7 +61,7 @@
 	localedef -i ${1} -f UTF-8 ${1}.UTF-8
 	if [ -f /lib/services/setup-nutyx/setup-nutyx.locales.gz ];then
 		bsdtar -xf /lib/services/setup-nutyx/setup-nutyx.locales.gz \
-		-C $TD lib/{setup-nutyx,xdg-user-dirs,dialog}.${L}
+		-C $TD lib/{setup-nutyx,xdg-user-dirs,dialog,util-linux}.${L}
 	fi
 	if [ -f $TD/lib/setup-nutyx.${L} ]; then
 		TEXTDOMAINFILE1=$TEXTDOMAINDIR/${L}/LC_MESSAGES/setup-nutyx.mo
@@ -82,7 +82,7 @@
 		-o $TEXTDOMAINFILE3
 	fi
 	if [ -f $TD/lib/util-linux.${L} ]; then
-		TEXTDOMAINFILE3=$TEXTDOMAINDIR/${L}/LC_MESSAGES/util-linux.mo
+		TEXTDOMAINFILE4=$TEXTDOMAINDIR/${L}/LC_MESSAGES/util-linux.mo
 		mkdir -p $TEXTDOMAINDIR/${L}/LC_MESSAGES
 		msgfmt $TD/lib/util-linux.${L} \
 		-o $TEXTDOMAINFILE4
@@ -1607,8 +1607,8 @@
 	do
 		[ ! -d ${2}/$DIR ] &&	mkdir -v ${2}/$DIR &> /dev/tty2
 	done
-	[ -a ${2}/dev/console ] || mknod -vm 640 ${2}/dev/console c 5 1 &> /dev/tty2
-	[ -a ${2}/dev/null ] || mknod -vm 664 ${2}/dev/null    c 1 3 &> /dev/tty2
+	[ -a ${2}/dev/console ] || mknod -m 640 ${2}/dev/console c 5 1 &> /dev/tty2
+	[ -a ${2}/dev/null ] || mknod -m 664 ${2}/dev/null    c 1 3 &> /dev/tty2
 	chmod -v 1777 ${2}/tmp &> /dev/tty2
 	echo "100") |
 	dialog --title "$(gettext "Install NuTyX")" --gauge "$(gettext "Please wait") ..." 6 50
