commit 00a634f9440db4254d0895f29984dc7b5a38d33b
Author: tnut <tnut@nutyx.org>
Date:   Thu Nov 26 15:37:31 2020 +0100

    install-nutyx,
    need bzip2 as well in a minimal install

diff --git a/scripts/install-nutyx.in b/scripts/install-nutyx.in
index 5ef5eb9a4..e885a010b 100755
--- a/scripts/install-nutyx.in
+++ b/scripts/install-nutyx.in
@@ -1128,7 +1128,7 @@ check_root() {
 }
 check_variables() {
 
-CR="nutyx busybox curl gcc glibc \
+CR="nutyx busybox curl gcc bzip2 glibc \
 zlib lz4 xz zstd libarchive openssl cards"
 
 if [ -z "$CHROOT" ]; then
