diff --git a/scripts/setup-nutyx.in b/scripts/setup-nutyx.in
index ff54e1f..c27df40 100755
--- a/scripts/setup-nutyx.in
+++ b/scripts/setup-nutyx.in
@@ -1240,7 +1240,8 @@ case $REPLY in
 	fr|fr-*|azerty)
 		echo '  Option "XkbLayout" "fr"' >> ${XORGKB}
 		echo 'export XKB_DEFAULT_LAYOUT="fr"' > ${WAYLKB};;
-	fr_CH*) echo '  Option "XkbLayout" "ch(fr)"' >> ${XORGKB}
+	fr_CH*) echo '  Option "XkbLayout" "ch"
+  Option "XkbVariant" "fr"' >> ${XORGKB}
 		echo 'export XKB_DEFAULT_LAYOUT="ch(fr)"' > ${WAYLKB};;
 	de_CH*) echo '  Option "XkbLayout" "ch"' >> ${XORGKB}
 		echo 'export XKB_DEFAULT_LAYOUT="ch"' > ${WAYLKB};;
