diff --git a/scripts/install-nutyx.in b/scripts/install-nutyx.in
index 4dfdcb2..63e6b28 100755
--- a/scripts/install-nutyx.in
+++ b/scripts/install-nutyx.in
@@ -99,9 +99,11 @@ print_error_msg() {
 unmountall() {
 if [ -d "$DEPOT_CD/depot" ]; then
 	umount ${LFS}/${DEPOT}
-	umount  $(cat /tmp/depot)
+	umount $(cat /tmp/depot)
 fi
 [ ! -z $MOUNT ] && umount ${LFS}/${DEPOT}
+[ -d /sys/firmware/efi/efivars ] && \
+umount ${LFS}/sys/firmware/efi/efivars
 
 umount ${LFS}/run
 umount ${LFS}/dev/shm
@@ -138,6 +140,8 @@ if [ ! -L ${LFS}/bin/sh ];then
   ln -sv ../sbin/busybox ${LFS}/bin/bash
   ln -sv bash ${LFS}/bin/sh
 fi
+[ -d /sys/firmware/efi/efivars ] && \
+mount -t efivarfs efivarfs ${LFS}/sys/firmware/efi/efivars
 }
 enter_chroot() {
 echo "
diff --git a/scripts/setup-nutyx.in b/scripts/setup-nutyx.in
index 498204e..c062759 100755
--- a/scripts/setup-nutyx.in
+++ b/scripts/setup-nutyx.in
@@ -222,6 +222,8 @@ setup_user()
 		if [ "$NAME" == "root" ]; then
 			rm -d /home
 			setup_password
+			SETUP_CONFIG_USER="no"
+			return
 		else
 			if [ "$RESULT" != "user" ] &&  [ "$RESULT" != "" ];then
 				RESULT=""
