diff --git a/po/setup-nutyx/ar.po b/po/setup-nutyx/ar.po
index 1900915..7364f55 100644
--- a/po/setup-nutyx/ar.po
+++ b/po/setup-nutyx/ar.po
@@ -484,18 +484,5 @@ msgstr "Advanced"
 msgid "Main"
 msgstr "Main"
 
-msgid "Choose your favorite Boot Manager"
-msgstr "Choose your favorite Boot Manager"
-
-msgid "Grub, Efibootmgr or nothing"
-msgstr "Grub, Efibootmgr or nothing"
-
-msgid "Advanced Linux Kernel boot manager"
-msgstr "Advanced Linux Kernel boot manager"
-
-msgid "Classic boot tool"
-msgstr "Classic boot tool"
-
-msgid "To use the one used on other distro"
-msgstr "To use the one used on other distro"
-
+msgid "Password is empty, please try again"
+msgstr "Password is empty, please try again"
diff --git a/po/setup-nutyx/da.po b/po/setup-nutyx/da.po
index 6862f10..612a0cb 100644
--- a/po/setup-nutyx/da.po
+++ b/po/setup-nutyx/da.po
@@ -450,17 +450,5 @@ msgstr "Fremskreden"
 msgid "Main"
 msgstr "Hovedmenu"
 
-msgid "Choose your favorite Boot Manager"
-msgstr "Choose your favorite Boot Manager"
-
-msgid "Grub, Efibootmgr or nothing"
-msgstr "Grub, Efibootmgr or nothing"
-
-msgid "Advanced Linux Kernel boot manager"
-msgstr "Advanced Linux Kernel boot manager"
-
-msgid "Classic boot tool"
-msgstr "Classic boot tool"
-
-msgid "To use the one used on other distro"
-msgstr "To use the one used on other distro"
+msgid "Password is empty, please try again"
+msgstr "Adgangskoden er tom, prøv venligst igen"
diff --git a/po/setup-nutyx/de.po b/po/setup-nutyx/de.po
index 62f6d74..2b9ebc4 100644
--- a/po/setup-nutyx/de.po
+++ b/po/setup-nutyx/de.po
@@ -486,17 +486,5 @@ msgstr "Erweiter"
 msgid "Main"
 msgstr "Haupt"
 
-msgid "Choose your favorite Boot Manager"
-msgstr "Choose your favorite Boot Manager"
-
-msgid "Grub, Efibootmgr or nothing"
-msgstr "Grub, Efibootmgr or nothing"
-
-msgid "Advanced Linux Kernel boot manager"
-msgstr "Advanced Linux Kernel boot manager"
-
-msgid "Classic boot tool"
-msgstr "Classic boot tool"
-
-msgid "To use the one used on other distro"
-msgstr "To use the one used on other distro"
+msgid "Password is empty, please try again"
+msgstr "Passwort ist leer, bitte versuche es noch einmal"
diff --git a/po/setup-nutyx/es.po b/po/setup-nutyx/es.po
index 3200036..54cd576 100644
--- a/po/setup-nutyx/es.po
+++ b/po/setup-nutyx/es.po
@@ -451,17 +451,5 @@ msgstr "Avanzado"
 msgid "Main"
 msgstr "Principal"
 
-msgid "Choose your favorite Boot Manager"
-msgstr "Choose your favorite Boot Manager"
-
-msgid "Grub, Efibootmgr or nothing"
-msgstr "Grub, Efibootmgr or nothing"
-
-msgid "Advanced Linux Kernel boot manager"
-msgstr "Advanced Linux Kernel boot manager"
-
-msgid "Classic boot tool"
-msgstr "Classic boot tool"
-
-msgid "To use the one used on other distro"
-msgstr "To use the one used on other distro"
+msgid "Password is empty, please try again"
+msgstr "La contraseña está vacía, inténtelo de nuevo."
diff --git a/po/setup-nutyx/fi.po b/po/setup-nutyx/fi.po
index 11e12bc..b25d719 100644
--- a/po/setup-nutyx/fi.po
+++ b/po/setup-nutyx/fi.po
@@ -483,17 +483,5 @@ msgstr "Lisävalikko"
 msgid "Main"
 msgstr "Päävalikko"
 
-msgid "Choose your favorite Boot Manager"
-msgstr "Choose your favorite Boot Manager"
-
-msgid "Grub, Efibootmgr or nothing"
-msgstr "Grub, Efibootmgr or nothing"
-
-msgid "Advanced Linux Kernel boot manager"
-msgstr "Advanced Linux Kernel boot manager"
-
-msgid "Classic boot tool"
-msgstr "Classic boot tool"
-
-msgid "To use the one used on other distro"
-msgstr "To use the one used on other distro"
+msgid "Password is empty, please try again"
+msgstr "Salasana on tyhjä, yritä uudelleen"
diff --git a/po/setup-nutyx/fr.po b/po/setup-nutyx/fr.po
index 98677d8..33851a3 100644
--- a/po/setup-nutyx/fr.po
+++ b/po/setup-nutyx/fr.po
@@ -486,17 +486,5 @@ msgstr "Avancé"
 msgid "Main"
 msgstr "Principal"
 
-msgid "Choose your favorite Boot Manager"
-msgstr "Choisissez votre gestionnaire d'amorçage favori"
-
-msgid "Grub, Efibootmgr or nothing"
-msgstr "Grub, Efibootmgr ou aucun des deux"
-
-msgid "Advanced Linux Kernel boot manager"
-msgstr "Gestionnaire d'amorçage avancé du kernel Linux"
-
-msgid "Classic boot tool"
-msgstr "Outil d'amorçage classique"
-
-msgid "To use the one used on other distro"
-msgstr "Amorçage depuis une autre distribution"
+msgid "Password is empty, please try again"
+msgstr "Le mot de passe est vide, essayez à nouveau svp"
diff --git a/po/setup-nutyx/it.po b/po/setup-nutyx/it.po
index 11e4e73..cb4db5a 100644
--- a/po/setup-nutyx/it.po
+++ b/po/setup-nutyx/it.po
@@ -451,17 +451,5 @@ msgstr "Avanzato"
 msgid "Main"
 msgstr "Principale"
 
-msgid "Choose your favorite Boot Manager"
-msgstr "Choose your favorite Boot Manager"
-
-msgid "Grub, Efibootmgr or nothing"
-msgstr "Grub, Efibootmgr or nothing"
-
-msgid "Advanced Linux Kernel boot manager"
-msgstr "Advanced Linux Kernel boot manager"
-
-msgid "Classic boot tool"
-msgstr "Classic boot tool"
-
-msgid "To use the one used on other distro"
-msgstr "To use the one used on other distro"
+msgid "Password is empty, please try again"
+msgstr "La password è vuota, riprova"
diff --git a/po/setup-nutyx/nl.po b/po/setup-nutyx/nl.po
index b46ee1c..02c717f 100644
--- a/po/setup-nutyx/nl.po
+++ b/po/setup-nutyx/nl.po
@@ -419,12 +419,12 @@ msgstr "Configure automatic update"
 
 msgid "Would you like to enable automatic update"
 msgstr "Would you like to enable automatic update"
-
-msgid "Make sure you reserve the first partition for ESP"
-msgstr "Make sure you reserve the first partition for ESP"
-
-msgid "At partitioning, declare the first partition as EFI System"
-msgstr "At partitioning, declare the first partition as EFI System"
+
+msgid "Make sure you reserve the first partition for ESP"
+msgstr "Make sure you reserve the first partition for ESP"
+
+msgid "At partitioning, declare the first partition as EFI System"
+msgstr "At partitioning, declare the first partition as EFI System"
 
 msgid "EFI partition not found"
 msgstr "EFI partition not found"
@@ -434,9 +434,9 @@ msgstr "GPT partition table not found"
 
 msgid "This installation won't boot again ..."
 msgstr "This installation won't boot again ..."
-
-msgid "You can remove the ISO media now."
-msgstr "U kunt de media met de ISO nu verwijderen."
+
+msgid "You can remove the ISO media now."
+msgstr "U kunt de media met de ISO nu verwijderen."
 
 msgid "Advanced Menu"
 msgstr "Geavanceerd menu"
@@ -446,18 +446,6 @@ msgstr "Geavanceerd"
 
 msgid "Main"
 msgstr "Hoofd"
-
-msgid "Choose your favorite Boot Manager"
-msgstr "Choose your favorite Boot Manager"
-
-msgid "Grub, Efibootmgr or nothing"
-msgstr "Grub, Efibootmgr or nothing"
-
-msgid "Advanced Linux Kernel boot manager"
-msgstr "Advanced Linux Kernel boot manager"
-
-msgid "Classic boot tool"
-msgstr "Classic boot tool"
-
-msgid "To use the one used on other distro"
-msgstr "To use the one used on other distro"
+
+msgid "Password is empty, please try again"
+msgstr "Wachtwoord is leeg, probeer het opnieuw"
diff --git a/po/setup-nutyx/nn.po b/po/setup-nutyx/nn.po
index 62c98c1..0e03a9a 100644
--- a/po/setup-nutyx/nn.po
+++ b/po/setup-nutyx/nn.po
@@ -483,17 +483,5 @@ msgstr "Avansert"
 msgid "Main"
 msgstr "Hoved"
 
-msgid "Choose your favorite Boot Manager"
-msgstr "Choose your favorite Boot Manager"
-
-msgid "Grub, Efibootmgr or nothing"
-msgstr "Grub, Efibootmgr or nothing"
-
-msgid "Advanced Linux Kernel boot manager"
-msgstr "Advanced Linux Kernel boot manager"
-
-msgid "Classic boot tool"
-msgstr "Classic boot tool"
-
-msgid "To use the one used on other distro"
-msgstr "To use the one used on other distro"
+msgid "Password is empty, please try again"
+msgstr "Passordet er tomt. Prøv igjen"
diff --git a/po/setup-nutyx/pt.po b/po/setup-nutyx/pt.po
index d38e69f..f449b10 100644
--- a/po/setup-nutyx/pt.po
+++ b/po/setup-nutyx/pt.po
@@ -450,17 +450,5 @@ msgstr "Avançado"
 msgid "Main"
 msgstr "Principal"
 
-msgid "Choose your favorite Boot Manager"
-msgstr "Choose your favorite Boot Manager"
-
-msgid "Grub, Efibootmgr or nothing"
-msgstr "Grub, Efibootmgr or nothing"
-
-msgid "Advanced Linux Kernel boot manager"
-msgstr "Advanced Linux Kernel boot manager"
-
-msgid "Classic boot tool"
-msgstr "Classic boot tool"
-
-msgid "To use the one used on other distro"
-msgstr "To use the one used on other distro"
+msgid "Password is empty, please try again"
+msgstr "A senha está vazia, por favor tente novamente"
diff --git a/po/setup-nutyx/ru.po b/po/setup-nutyx/ru.po
index 579a7f9..b7bb404 100644
--- a/po/setup-nutyx/ru.po
+++ b/po/setup-nutyx/ru.po
@@ -388,10 +388,10 @@ msgid "Swap partition not found"
 msgstr "Swap partition not found"
 
 msgid "recommended"
-msgstr "recommended"
+msgstr "рекомендуемые"
 
 msgid "skilled"
-msgstr "skilled"
+msgstr "опытный"
 
 msgid "Configure the wireless access point"
 msgstr "Configure the wireless access point"
@@ -406,7 +406,7 @@ msgid "Enter the password of"
 msgstr "Enter the password of"
 
 msgid "File"
-msgstr "File"
+msgstr "Файл"
 
 msgid "successfully created"
 msgstr "successfully created"
@@ -436,7 +436,7 @@ msgid "GPT partition table not found"
 msgstr "GPT partition table not found"
 
 msgid "This installation won't boot again ..."
-msgstr "This installation won't boot again ..."
+msgstr "Эта установка больше не загружается ..."
 
 msgid "You can remove the ISO media now."
 msgstr "Теперь вы можете удалить носитель, содержащий ISO."
@@ -445,22 +445,10 @@ msgid "Advanced Menu"
 msgstr "Расширенное меню"
 
 msgid "Advanced"
-msgstr "Gelismis"
+msgstr "Передовой"
 
 msgid "Main"
-msgstr "Ana"
+msgstr "Главный"
 
-msgid "Choose your favorite Boot Manager"
-msgstr "Choose your favorite Boot Manager"
-
-msgid "Grub, Efibootmgr or nothing"
-msgstr "Grub, Efibootmgr or nothing"
-
-msgid "Advanced Linux Kernel boot manager"
-msgstr "Advanced Linux Kernel boot manager"
-
-msgid "Classic boot tool"
-msgstr "Classic boot tool"
-
-msgid "To use the one used on other distro"
-msgstr "To use the one used on other distro"
+msgid "Password is empty, please try again"
+msgstr "Пароль пуст, попробуйте еще раз"
diff --git a/po/setup-nutyx/sv.po b/po/setup-nutyx/sv.po
index 1a030b7..721547d 100644
--- a/po/setup-nutyx/sv.po
+++ b/po/setup-nutyx/sv.po
@@ -486,17 +486,5 @@ msgstr "Avancerad"
 msgid "Main"
 msgstr "Huvud"
 
-msgid "Choose your favorite Boot Manager"
-msgstr "Choose your favorite Boot Manager"
-
-msgid "Grub, Efibootmgr or nothing"
-msgstr "Grub, Efibootmgr or nothing"
-
-msgid "Advanced Linux Kernel boot manager"
-msgstr "Advanced Linux Kernel boot manager"
-
-msgid "Classic boot tool"
-msgstr "Classic boot tool"
-
-msgid "To use the one used on other distro"
-msgstr "To use the one used on other distro"
+msgid "Password is empty, please try again"
+msgstr "Lösenordet är tomt, försök igen"
diff --git a/po/setup-nutyx/tr.po b/po/setup-nutyx/tr.po
index 34d00bb..b65d74d 100644
--- a/po/setup-nutyx/tr.po
+++ b/po/setup-nutyx/tr.po
@@ -450,17 +450,5 @@ msgstr "Advanced"
 msgid "Main"
 msgstr "Main"
 
-msgid "Choose your favorite Boot Manager"
-msgstr "Choose your favorite Boot Manager"
-
-msgid "Grub, Efibootmgr or nothing"
-msgstr "Grub, Efibootmgr or nothing"
-
-msgid "Advanced Linux Kernel boot manager"
-msgstr "Advanced Linux Kernel boot manager"
-
-msgid "Classic boot tool"
-msgstr "Classic boot tool"
-
-msgid "To use the one used on other distro"
-msgstr "To use the one used on other distro"
+msgid "Password is empty, please try again"
+msgstr "Şifre boş, lütfen tekrar deneyin"
diff --git a/scripts/install-nutyx.in b/scripts/install-nutyx.in
index 6bf961a..75a4755 100755
--- a/scripts/install-nutyx.in
+++ b/scripts/install-nutyx.in
@@ -1139,6 +1139,8 @@ local FILE
 [ ! -f "$(which xorriso)" ] && error "You need to install 'libisoburn'"
 [ ! -f "$(which isohybrid)" ] && error "You need to install 'syslinux'"
 [ ! -f "$(which fatlabel)" ] && error "You need to install 'dosfstools'"
+[ ! -f "$(which mksquashfs)" ] && error "You need to install 'squashfs'"
+
 for FILE in libutil.c32 libcom32.c32 ldlinux.c32 vesamenu.c32 isolinux.bin
 do
 	[ ! -f /usr/lib/syslinux/$FILE ] && \
@@ -1215,6 +1217,7 @@ simple-scan gnome-weather gnome-chess balsa"
 [ "$KERNEL" = "NONE" ] && KERNEL="none"
 
 if [ "$KERNEL" != "none"  ]; then
+	[ -z $ISO ] || KERNEL=""
 	CORE="acl attr bash coreutils readline \
 expat ncurses lzo gmp flex file gzip mpc mpfr \
 grep sed findutils binutils \
@@ -1260,9 +1263,7 @@ if [ ! -z "$ISO" ];then
 for k in \$(ls /lib/modules/)
 do
   mkinitramfs \$k
-done
-
-sed -i \"s@kernel-419@\$KERNEL@g\" /sbin/setup-nutyx"
+done"
 	if [ -z $MOUNT ]; then
 		DEVEL+="
 cards purge"
@@ -1330,7 +1331,7 @@ SET_WCOL="\\033[${WCOL}G"    # at the $WCOL char
 CURS_UP="\\033[1A\\033[0G"   # Up one line, at the 0'th char
 
 ## Set color commands, used via $ECHO
-# Please consult `man console_codes for more information
+# Please consult man console_codes for more information
 # under the "ECMA-48 Set Graphics Rendition" section
 #
 # Warning: when switching from a 8bit to a 9bit font,
@@ -1389,24 +1390,23 @@ make_iso_directories() {
 default install
 label install
  menu label ^Install
- kernel /boot/kernel-#VERSION_KERNEL#
- append initrd=/boot/initrd-#VERSION_KERNEL# vga=788 ro quiet
+ kernel /boot/bzImage
+ append initrd=/boot/initrd vga=788 ro quiet
 
 label install-debug
  menu label install-^Debug
- kernel /boot/kernel-#VERSION_KERNEL#
- append initrd=/boot/initrd-#VERSION_KERNEL#  vga=788 ro rootdelay=8 testiso
+ kernel /boot/bzImage
+ append initrd=/boot/initrd vga=788 ro rootdelay=8 testiso
 
 label install-toram
  menu label Install copied to ^RAM
- kernel /boot/kernel-#VERSION_KERNEL#
- append initrd=/boot/initrd-#VERSION_KERNEL# vga=788 ro quiet toram
+ kernel /boot/bzImage
+ append initrd=/boot/initrd vga=788 ro quiet toram
 
 label install-debug-toram
  menu label install-Debug copied to RA^M
- kernel /boot/kernel-#VERSION_KERNEL#
- append initrd=/boot/initrd-#VERSION_KERNEL#  vga=788 ro rootdelay=8 testiso toram
-fi
+ kernel /boot/bzImage
+ append initrd=/boot/initrd vga=788 ro rootdelay=8 testiso toram
 
 EOF
 	cat > ${LFS}/ISO/isolinux/isolinux.cfg << "EOF"
@@ -1420,30 +1420,30 @@ EOF
 	cat > $TMP/live.cfg << "EOF"
 label live
  menu label ^Live
- kernel /boot/kernel-#VERSION_KERNEL#
- append initrd=/boot/initrd-#VERSION_KERNEL# vga=788 ro quiet live
+ kernel /boot/bzImage
+ append initrd=/boot/initrd vga=788 ro quiet live
 
 label live-debug
  menu label li^ve-Debug
- kernel /boot/kernel-#VERSION_KERNEL#
- append initrd=/boot/initrd-#VERSION_KERNEL# vga=788 ro rootdelay=8 testiso live
+ kernel /boot/bzImage
+ append initrd=/boot/initrd vga=788 ro rootdelay=8 testiso live
 
 label live-toram
  menu label live copied to ^RAM
- kernel /boot/kernel-#VERSION_KERNEL#
- append initrd=/boot/initrd-#VERSION_KERNEL# vga=788 ro quiet live toram
+ kernel /boot/bzImage
+ append initrd=/boot/initrd vga=788 ro quiet live toram
 
 label live-debug-toram
  menu label live-Debug copied to ^RAM
- kernel /boot/kernel-#VERSION_KERNEL#
- append initrd=/boot/initrd-#VERSION_KERNEL# vga=788 ro rootdelay=8 testiso live toram
+ kernel /boot/bzImage
+ append initrd=/boot/initrd vga=788 ro rootdelay=8 testiso live toram
 
 EOF
 	cat > $TMP/menu.cfg << "EOF"
 menu hshift 7
 menu width 80
 
-menu title NuTyX #VERSION_NUTYX# #VERSION_KERNEL#
+menu title NuTyX #VERSION_NUTYX#
 include stdmenu.cfg
 include install.cfg
 include live.cfg
@@ -1594,7 +1594,7 @@ ISO Filename $LFS/NuTyX_${ARCH}-$VERSION_NUTYX-$ISO.iso"
 }
 print_help() {
 	echo "
-usage: `basename ${INSTALL_COMMAND}` [options]
+usage: $(sename ${INSTALL_COMMAND}) [options]
 options
 
   -ec,  --enter-chroot    will enter in the chroot if exist
@@ -1670,23 +1670,10 @@ SCRIPTDIR=$(pwd -P)
 
 cd $LFS
 
-if [ -z $KERNEL ];then
-  VERSION_KERNEL="$(basename lib/modules/*)"
-  cp -v boot/bzImage ISO/boot/kernel-${VERSION_KERNEL}
-  cp -v boot/initrd-${VERSION_KERNEL} ISO/boot/
-else
-  for VERSION_KERNEL in $(ls lib/modules/)
-  do
-    MK="${VERSION_KERNEL/NuTyX/kernel}"
-    MK="${MK/$KERNEL/}"
-    [ -z "$(echo $MK| cut -d "-" -f2)" ] && \
-    break
-  done
-  cp -v boot/$KERNEL ISO/boot/kernel-$VERSION_KERNEL
-  cp -v boot/initrd-$VERSION_KERNEL ISO/boot/initrd-$VERSION_KERNEL
-fi
-
+VERSION_KERNEL="$(basename lib/modules/*)"
 
+cp -v boot/bzImage ISO/boot/bzImage
+cp -v boot/initrd-${VERSION_KERNEL} ISO/boot/initrd
 
 ISO="NuTyX_${ARCH}-$VERSION_NUTYX-$ISO"
 cd ${ISODIR} || error "cd ${ISODIR}"
@@ -1699,13 +1686,10 @@ done
 sed "s|#VERSION_NUTYX#|${VERSION_NUTYX}-$ISO|" \
 	$TMP/menu.cfg \
 	> isolinux/menu.cfg
-sed "s|#VERSION_KERNEL#|${VERSION_KERNEL}|" \
-	$TMP/live.cfg \
-	> isolinux/live.cfg
-sed "s|#VERSION_KERNEL#|${VERSION_KERNEL}|" \
-	$TMP/install.cfg \
-	> isolinux/install.cfg
-sed -i "s|#VERSION_KERNEL#|${VERSION_KERNEL}|" isolinux/menu.cfg
+cp $TMP/live.cfg \
+	isolinux/live.cfg
+cp $TMP/install.cfg \
+	isolinux/install.cfg
 
 # efi stufs
 rm -rf iso
@@ -1752,45 +1736,44 @@ else
   set color_highlight=white/blue
 fi
 
-menuentry "#VERSION_NUTYX# #VERSION_KERNEL# (UEFI) normal" {
+menuentry "#VERSION_NUTYX# (UEFI) normal" {
 
-    linux /boot/kernel-#VERSION_KERNEL#  ro quiet
-    initrd /boot/initrd-#VERSION_KERNEL#
+    linux /boot/bzImage ro quiet
+    initrd /boot/initrd
 }
-menuentry "#VERSION_NUTYX# #VERSION_KERNEL# (UEFI) debug" {
+menuentry "#VERSION_NUTYX# (UEFI) debug" {
     set background_color=black
-    linux /boot/kernel-#VERSION_KERNEL#  ro testiso rootdelay=8
-    initrd /boot/initrd-#VERSION_KERNEL#
+    linux /boot/bzImage ro testiso rootdelay=8
+    initrd /boot/initrd
 }
-menuentry "#VERSION_NUTYX# #VERSION_KERNEL# (UEFI) copied to RAM" {
+menuentry "#VERSION_NUTYX# (UEFI) copied to RAM" {
 
-    linux /boot/kernel-#VERSION_KERNEL#  ro quiet toram
-    initrd /boot/initrd-#VERSION_KERNEL#
+    linux /boot/bzImage ro quiet toram
+    initrd /boot/initrd
 }
-menuentry "#VERSION_NUTYX# #VERSION_KERNEL# (UEFI) debug copied to RAM" {
+menuentry "#VERSION_NUTYX# (UEFI) debug copied to RAM" {
     set background_color=black
-    linux /boot/kernel-#VERSION_KERNEL#  ro testiso rootdelay=8 toram
-    initrd /boot/initrd-#VERSION_KERNEL#
+    linux /boot/bzImage ro testiso rootdelay=8 toram
+    initrd /boot/initrd
 }
-menuentry "#VERSION_NUTYX# #VERSION_KERNEL# (UEFI) installable live" {
-    linux /boot/kernel-#VERSION_KERNEL#  ro quiet rootdelay=5 live
-    initrd /boot/initrd-#VERSION_KERNEL#
+menuentry "#VERSION_NUTYX# (UEFI) installable live" {
+    linux /boot/bzImage ro quiet rootdelay=5 live
+    initrd /boot/initrd
 }
-menuentry "#VERSION_NUTYX# #VERSION_KERNEL# (UEFI) debug live" {
-    linux /boot/kernel-#VERSION_KERNEL#  ro testiso rootdelay=8 live
-    initrd /boot/initrd-#VERSION_KERNEL#
+menuentry "#VERSION_NUTYX# (UEFI) debug live" {
+    linux /boot/bzImage ro testiso rootdelay=8 live
+    initrd /boot/initrd
 }
-menuentry "#VERSION_NUTYX# #VERSION_KERNEL# (UEFI) installable live copied to RAM" {
-    linux /boot/kernel-#VERSION_KERNEL#  ro quiet rootdelay=5 live toram
-    initrd /boot/initrd-#VERSION_KERNEL#
+menuentry "#VERSION_NUTYX# (UEFI) installable live copied to RAM" {
+    linux /boot/bzImage ro quiet rootdelay=5 live toram
+    initrd /boot/initrd
 }
-menuentry "#VERSION_NUTYX# #VERSION_KERNEL# (UEFI) debug live copied to RAM" {
-    linux /boot/kernel-#VERSION_KERNEL#  ro testiso rootdelay=8 live toram
-    initrd /boot/initrd-#VERSION_KERNEL#
+menuentry "#VERSION_NUTYX# (UEFI) debug live copied to RAM" {
+    linux /boot/bzImage ro testiso rootdelay=8 live toram
+    initrd /boot/initrd
 }
 EOF
 sed -i "s|#VERSION_NUTYX#|${VERSION_NUTYX}-$ISO|" boot/grub/grub.cfg
-sed -i "s|#VERSION_KERNEL#|${VERSION_KERNEL}|" boot/grub/grub.cfg
 sed -i "s|#ARCH#|${ARCH}|" boot/grub/grub.cfg
 
 for FILE in kernel initrd
diff --git a/scripts/setup-nutyx.in b/scripts/setup-nutyx.in
index ee8918c..e03e809 100755
--- a/scripts/setup-nutyx.in
+++ b/scripts/setup-nutyx.in
@@ -182,10 +182,14 @@ setup_password()
 		--insecure "$@" \
 		--passwordbox "$(gettext "Confirm the new password"):" 8 60 2>&1 1>&3`
 
-		if [ "$PASSWORD" == "$CONFIRM_PASSWORD" ]; then
-			RESULT="true"
-		else
-			dialog --msgbox "$(gettext "Passwords are differents, please try again")" 5 60
+    if [ "$PASSWORD" == "" ] || [ "$CONFIRM_PASSWORD" == "" ]; then
+			dialog --msgbox "$(gettext "Password is empty, please try again")" 5 60
+    else
+			if [ "$PASSWORD" == "$CONFIRM_PASSWORD" ]; then
+				RESULT="true"
+			else
+				dialog --msgbox "$(gettext "Passwords are differents, please try again")" 5 60
+			fi
 		fi
 	done
 
@@ -1705,16 +1709,13 @@ install_nutyx() {
 		cp -av /etc/locale ${2}/usr/lib/ &> /dev/tty3
 	fi
 	# Kernels
-	cp -av ${1}/boot/kernel* ${2}/boot/ &> /dev/tty3
-	for i in /lib/modules/*
-	do
-		VERSNAME="$(basename $i)"
-		EXT="-${VERSNAME/*-/}"
-		[ "$EXT" == "-NuTyX" ] && EXT=""
-		mv ${2}/boot/kernel-${VERSNAME} ${2}/boot/bzImage${EXT}
-		ln -svf bzImage${EXT} ${2}/boot/kernel-${VERSNAME} &> /dev/tty3
-		ln -svf bzImage${EXT} ${2}/boot/kernel${EXT} &> /dev/tty3
-	done
+	cp -av ${1}/boot/bzImage ${2}/boot/ &> /dev/tty3
+	cp -av ${1}/boot/initrd-* ${2}/boot/initrd &> /dev/tty3
+
+
+	ln -svf bzImage ${2}/boot/kernel-stable &> /dev/tty3
+	ln -svf bzImage ${2}/boot/kernel &> /dev/tty3
+
 	if  [ -f ${1}/boot/grub/splash.png ]; then
 		mkdir -pv ${2}/boot/grub/ &> /dev/tty3
 		cp -v ${1}/boot/grub/splash.png ${2}/boot/grub/ &> /dev/tty3
@@ -2029,72 +2030,14 @@ PATH=$PATH" >> ${GRUB_LOG_FILE}
 		BOOT_FS="$(print_file_system $PARTITION)"
 		NuTyX_FS="$(print_file_system $NuTyX_PART)"
 		echo "
-submenu \"NuTyX SysV \" {
- submenu \" NuTyX SysV Quiet Mode \" {
-  submenu \"  NuTyX SysV Quiet Latest version \" {
+menuentry \"NuTyX SysV Quiet Mode\" {
    linux /${GRUB_DESTINATION_DIR}kernel root=${NuTyX_PART} ro quiet verbose=no
    #  initrd /${GRUB_DESTINATION_DIR}initrd
-  }
-  submenu \"  NuTyX SysV Quiet LTS version \" {
-   linux /${GRUB_DESTINATION_DIR}kernel-419 root=${NuTyX_PART} ro quiet verbose=no
-   #  initrd /${GRUB_DESTINATION_DIR}initrd-419
-  }
- }
- submenu \" NuTyX SysV Verbose Mode \" {
-  submenu \"  NuTyX SysV Verbose Latest version \" {
-   linux /${GRUB_DESTINATION_DIR}kernel root=${NuTyX_PART} ro
-   #  initrd /${GRUB_DESTINATION_DIR}initrd
-  }
-  submenu \"  NuTyX SysV Verbose LTS version \" {
-   linux /${GRUB_DESTINATION_DIR}kernel-419 root=${NuTyX_PART} ro
-   #  initrd /${GRUB_DESTINATION_DIR}initrd-419
-  }
- }
-}
-submenu \"NuTyX SystemD \" {
- submenu \" NuTyX SystemD Quiet Mode \" {
-  submenu \"  NuTyX SystemD Quiet Latest version \" {
-    linux /${GRUB_DESTINATION_DIR}kernel init=/sbin/systemd-init root=${NuTyX_PART} ro quiet verbose=no
-  #  initrd /${GRUB_DESTINATION_DIR}initrd
-  }
-  submenu \"  NuTyX SystemD Quiet LTS version \" {
-    linux /${GRUB_DESTINATION_DIR}kernel-419 init=/sbin/systemd-init root=${NuTyX_PART} ro quiet verbose=no
-  #  initrd /${GRUB_DESTINATION_DIR}initrd-419
-  }
- }
- submenu \" NuTyX SystemD Verbose Mode \" {
-  submenu \"  NuTyX SystemD Verbose Latest version \" {
-    linux /${GRUB_DESTINATION_DIR}kernel init=/sbin/systemd-init root=${NuTyX_PART} ro
-  #  initrd /${GRUB_DESTINATION_DIR}initrd
-  }
-  submenu \"  NuTyX SystemD Verbose LTS version \" {
-    linux /${GRUB_DESTINATION_DIR}kernel-419 init=/sbin/systemd-init root=${NuTyX_PART} ro
-  #  initrd /${GRUB_DESTINATION_DIR}initrd-419
-  }
- }
 }
-submenu \"NuTyX RuNyX (Experimental)\" {
- submenu \" NuTyX RuNyX Quiet Mode \" {
-  submenu \"  NuTyX RuNyX Quiet Latest version \" {
-   linux /${GRUB_DESTINATION_DIR}kernel init=/sbin/runit-init root=${NuTyX_PART} ro quiet verbose=no
+menuentry \"NuTyX SysV Verbose Mode \" {
+   linux /${GRUB_DESTINATION_DIR}kernel root=${NuTyX_PART} ro
    #  initrd /${GRUB_DESTINATION_DIR}initrd
-  }
-  submenu \"  NuTyX RuNyX Quiet LTS version \" {
-  linux /${GRUB_DESTINATION_DIR}kernel-419 init=/sbin/runit-init root=${NuTyX_PART} ro quiet verbose=no
-  #  initrd /${GRUB_DESTINATION_DIR}initrd-419
-  }
- }
-  submenu \" NuTyX RuNyX Verbose Mode \" {
-   submenu \"  NuTyX RuNyX Verbose Latest version \" {
-    linux /${GRUB_DESTINATION_DIR}kernel init=/sbin/runit-init root=${NuTyX_PART} ro
-    #  initrd /${GRUB_DESTINATION_DIR}initrd
-   }
-   submenu \"  NuTyX RuNyX Verbose LTS version \" {
-    linux /${GRUB_DESTINATION_DIR}kernel-419 init=/sbin/runit-init root=${NuTyX_PART} ro
-    #  initrd /${GRUB_DESTINATION_DIR}initrd-419
-   }
-  }
- }" >> ${GRUB_CONFIG_DIR}/${GRUB_CONFIG_FILE}
+}" >> ${GRUB_CONFIG_DIR}/${GRUB_CONFIG_FILE}
 	else
 		#
 		# NuTyX is not yet installed
@@ -2131,29 +2074,29 @@ submenu \"NuTyX RuNyX (Experimental)\" {
 #
 #menuentry  "Harddisk number 2 and partition number 3 verbose mode" {
 #   set root=(hd1,3)
-#   linux /boot/kernel-419 root=/dev/sdb3 ro
+#   linux /boot/kernel init=/sbin/init root=/dev/sdb3 ro
 #}
 #
 # The same one with systemd:
 #
 #menuentry  "Harddisk number 2 and partition number 3 verbose mode" {
 #   set root=(hd1,3)
-#   linux /boot/kernel-419 init=/sbin/systemd-init root=/dev/sdb3 ro
+#   linux /boot/kernel init=/sbin/systemd-init root=/dev/sdb3 ro
 #}
 #
 # The same one with RuNyX (Experimental):
 #
 #menuentry  "Harddisk number 2 and partition number 3" {
 #   set root=(hd1,3)
-#   linux /boot/kernel-419 init=/sbin/runit-init root=/dev/sdb3 ro quiet verbose=no
+#   linux /boot/kernel init=/sbin/runit-init root=/dev/sdb3 ro quiet verbose=no
 #}
 #
 # An other one with initrd and rootdelay (to use with USB stick for ex)
 #
 #menuentry  "Harddisk number 2 and partition number 1 with initrd" {
 #   set root=(hd1,1)
-#   linux /boot/kernel-419 init=/sbin/runit-init root=/dev/sdb1 ro rootdelay=5
-#   initrd /boot/initrd-419
+#   linux /boot/kernel init=/sbin/runit-init root=/dev/sdb1 ro rootdelay=5
+#   initrd /boot/initrd
 #}
 #
 EOF
