diff --git a/man/Makefile b/man/Makefile
index fdb9543..66f3f93 100644
--- a/man/Makefile
+++ b/man/Makefile
@@ -1,7 +1,7 @@
 #
 #  nutyx
 #
-#  Copyright (c) 2016 by NuTyX team (http://nutyx.org)
+#  Copyright (c) 2016 - 2021 by NuTyX team (http://nutyx.org)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
@@ -21,7 +21,7 @@
 
 include ../Makefile.inc
 
-MANPAGES = nutyx.1 ifup.8
+MANPAGES = nutyx.1 ifup.8 genfstab.8
 
 all: man mantxt
 
@@ -48,6 +48,7 @@ mantxt: man $(MANPAGES:=.txt)
 install: all
 	install -D -m0644 nutyx.1 $(MANDIR)/man1/nutyx.1
 	install -D -m0644 ifup.8 $(MANDIR)/man8/ifup.8
+	install -D -m0644 genfstab.8 $(MANDIR)/man8/genfstab.8
 
 clean:
 	rm -f $(MANPAGES)
diff --git a/scripts/install-nutyx.in b/scripts/install-nutyx.in
index 4709c46..4dfdcb2 100755
--- a/scripts/install-nutyx.in
+++ b/scripts/install-nutyx.in
@@ -1503,7 +1503,7 @@ else
 	[ -z $MOUNT ] && echo "cards sync" >> \
 	$LFS/$SETUP_FILE
 	echo "cards install -n $CORE $DEVEL
-which genfstab > /dev/null && genfstab >> etc/fstab
+which genfstab > /dev/null && genfstab >> /etc/fstab
 exit" >> $LFS/$SETUP_FILE
 fi
 chmod 177 $LFS/$SETUP_FILE
