diff --git a/scripts/setup-nutyx.in b/scripts/setup-nutyx.in
index d97e051..3646978 100755
--- a/scripts/setup-nutyx.in
+++ b/scripts/setup-nutyx.in
@@ -1395,6 +1395,7 @@ select_disk() {
 	fi
 }
 select_partition() {
+	echo "Entering select_partition()" > /dev/tty2
 	local retval result
 	retval=""
 	result=""
@@ -1425,6 +1426,7 @@ select_partition() {
 }
 check_available_disk() {
 	echo "Entering check_available_disk()" > /dev/tty2
+	CHECK_HD_OK="no"
 	unset DISK_LIST
 	local DISK_FULL_LIST
 	DISK_FULL_LIST=`lsblk -p -n -d -o NAME -e 1,3,7,11`
@@ -1447,7 +1449,8 @@ check_available_disk() {
 		fi
 		CHECK_HD_OK="yes"
 	fi
-	echo "DISK_LIST=$DISK_LIST (1450)" > /dev/tty2
+	echo "CHECK_HD_OK=$CHECK_HD_OK (1451)
+DISK_LIST=$DISK_LIST (1452)" > /dev/tty2
 }
 check_available_partition() {
 	echo "Entering check_available_partition()" > /dev/tty2
@@ -1455,10 +1458,7 @@ check_available_partition() {
 	unset PARTITION
 	unset CHECK_SINGLE_PARTITION
 	local PARTITION_FULL_LIST MEM_TOTAL SWAP
-	if [ "$CHECK_HD_OK" == "no" ]; then
-		check_available_disk
-		 [ "$CHECK_HD_OK" == "no" ] && return
-	fi
+
 	PARTITION_LIST=`for i in $DISK_LIST; do lsblk -p -l -n -o NAME $i| tail -n +2; done`
 	if [ ! -z "$PARTITION_LIST" ];then
 		CHECK_PART_OK="yes"
@@ -1580,6 +1580,7 @@ setup_partition() {
 	local retval result
 	check_available_disk
 	[ "$CHECK_HD_OK" == "no" ] && return
+
 	retval=""
 	if [ -z $DISK ]; then
 		select_disk "$(gettext "Create all your partitions")"
@@ -1609,11 +1610,8 @@ setup_partition() {
 setup_format() {
 	local retval FS
 	retval=""
-	if [ "$CHECK_HD_OK" == "no" ]; then
-		check_available_disk
-		[ "$CHECK_HD_OK" == "no" ] && return
-	fi
-	check_available_partition
+
+	[ "$CHECK_PART_OK" == "no" ] && check_available_partition
 	[ "$CHECK_PART_OK" == "no" ] && return
 	if [ -z "$PARTITION" ]; then
 		select_partition "$(gettext "Format a partition")"
@@ -1750,14 +1748,13 @@ install_nutyx() {
 setup_erase() {
 	local MF retval
 	retval=""
-	if [ "$CHECK_HD_OK" == "no" ]; then
-		check_available_disk
-		[ "$CHECK_HD_OK" == "no" ] && return
-	fi
-	if [ "$CHECK_PART_OK" == "no" ]; then
-		check_available_partition
-		[ "$CHECK_PART_OK" == "no" ] && return
-	fi
+
+	[ "$CHECK_HD_OK" == "yes" ] || check_available_disk
+	[ "$CHECK_HD_OK" == "yes" ] || return
+
+	[ "$CHECK_PART_OK" == "no" ] && check_available_partition
+	[ "$CHECK_PART_OK" == "no" ] && return
+
 	echo "PARTITION=$PARTITION (1706)" > /dev/tty2
 	echo "CHECK_SINGLE_PARTITION=$CHECK_SINGLE_PARTITION (1707)" > /dev/tty2
 	[ "$CHECK_SINGLE_PARTITION" == "no" ] && unset PARTITION
@@ -1779,23 +1776,36 @@ setup_erase() {
 }
 setup_install() {
 	echo "Entering setup_install()" > /dev/tty2
+
+	[ "$CHECK_HD_OK" == "yes" ] || check_available_disk
+	[ "$CHECK_HD_OK" == "yes" ] || return
+
 	local MF retval
 	retval=""
-  if [ "$CHECK_HD_OK" == "no" ]; then
-    check_available_disk
-    [ "$CHECK_HD_OK" == "no" ] && return
-  fi
-    check_available_partition
-    [ "$CHECK_PART_OK" == "no" ] && return
-	echo "PARTITION=$PARTITION (1736)" > /dev/tty2
-	echo "CHECK_SINGLE_PARTITION=$CHECK_SINGLE_PARTITION (1779)" > /dev/tty2
-	echo "CHECK_PART_OK=$CHECK_PART_OK (1738)" > /dev/tty2
+
+	echo "PARTITION=$PARTITION (1787)
+CHECK_SINGLE_PARTITION=$CHECK_SINGLE_PARTITION (1788)
+CHECK_PART_OK=$CHECK_PART_OK (1789)
+CHECK_HD_OK=$CHECK_HD_OK (1790)
+PARTITION=$PARTITION (1791)" > /dev/tty2
+
+	[ "$CHECK_PART_OK" == "no" ] && check_available_partition
+	[ "$CHECK_PART_OK" == "no" ] && return
+
 	[ "$CHECK_SINGLE_PARTITION" == "no" ] && unset PARTITION
+
+	echo "PARTITION=$PARTITION (1796)
+CHECK_SINGLE_PARTITION=$CHECK_SINGLE_PARTITION (1797)
+CHECK_PART_OK=$CHECK_PART_OK (1798)" > /dev/tty2
+
 	if [ -z $PARTITION ]; then
 		select_partition "$(gettext "Install NuTyX")"
+		[ "$PARTITION" == "no" ] && setup_format
 		[ "$PARTITION" == "no" ] && return
 	fi
-	echo "PARTITION=$PARTITION (1744)" > /dev/tty2
+
+	echo "PARTITION=$PARTITION (1806)" > /dev/tty2
+
 	if [ "$CHECK_PART_OK" == "new" ]; then
 		mkfs.ext4 -F $PARTITION > /dev/tty2 || error "mkfs.ext4 $PARTITION"
 		$SETUP_LOG "setup-nutyx" "mkfs.ext4 -F $PARTITION"
@@ -1805,15 +1815,15 @@ setup_install() {
 		fi
 	fi
 	FORMAT="$(print_file_system $PARTITION)"
-	echo "FORMAT $PARTITION=$FORMAT (1754)" > /dev/tty2
+	echo "FORMAT $PARTITION=$FORMAT (1816)" > /dev/tty2
 	if [ "$FORMAT" == "ntfs" ]; then
 		setup_partition
-		[ "$FORMAT" == "ntfs" ] && return
-	fi
-	if [ "$FORMAT" == "no" ]; then
-		setup_format
-		[ "$FORMAT" == "no" ] && return
+		return
 	fi
+
+	[ "$FORMAT" == "no" ] && setup_format
+	[ "$FORMAT" == "no" ] && return
+
 	SETUP_INSTALL="no"
 	MF=/mnt/hd
 	mkdir -p $MF
@@ -1830,7 +1840,7 @@ setup_install() {
 	SETUP_INSTALL="yes"
 	echo "$PARTITION" > /root/boot
 	echo "$PARTITION" > $TMP/boot
-	echo "CHECK_PART_OK=$CHECK_PART_OK (1775)" > /dev/tty2
+	echo "CHECK_PART_OK=$CHECK_PART_OK (1836)" > /dev/tty2
 	if [ "$CHECK_PART_OK" == "new" ]; then
 		setup_grub
 		return
@@ -1838,8 +1848,8 @@ setup_install() {
 	if [ -d "$EFIVARS" ];then
 		FORMAT="no"
 		[ ! -z $EFI_SYS_PART ] && FORMAT="$(print_file_system $EFI_SYS_PART)"
-		echo "EFI_SYS_PART=$EFI_SYS_PART (1791)" > /dev/tty2
-		echo "$EFI_SYS_PART FORMAT=$FORMAT (1792)" > /dev/tty2
+		echo "EFI_SYS_PART=$EFI_SYS_PART (1843)" > /dev/tty2
+		echo "$EFI_SYS_PART FORMAT=$FORMAT (1844)" > /dev/tty2
 		if [ "$FORMAT" == "vfat" ];then
 			dialog --title "   $(gettext "Configuration of the boot of the computer") (GRUB)   " \
 			--yesno "$(gettext "Configure the boot of the PC") ? " 0 0
@@ -1888,14 +1898,13 @@ setup_grub() {
 	GRUB_CONFIG_DIR=""
 	GRUB_DESTINATION_DIR=""
 	GRUB_LOG_FILE=$TMP/grub.log
-	if [ "$CHECK_HD_OK" == "no" ]; then
-		check_available_disk
-		[ "$CHECK_HD_OK" == "no" ] && return
-	fi
-	if [ "$CHECK_PART_OK" == "no" ]; then
-		check_available_partition
-		[ "$CHECK_PART_OK" == "no" ] && return
-	fi
+
+	[ "$CHECK_HD_OK" == "no" ] || check_available_disk
+	[ "$CHECK_HD_OK" == "no" ] || return
+
+	[ "$CHECK_PART_OK" == "no" ] && check_available_partition
+	[ "$CHECK_PART_OK" == "no" ] && return
+
 	if [ -z $DISK ]; then
 		select_disk "$(gettext "Configure the boot of the PC")"
 		[ "$DISK" == "no" ] && return
@@ -2142,8 +2151,6 @@ setup_advanced_initrd() {
 				fi
 				;;
 			"$(gettext "Install")")
-				echo "CHECK_SINGLE_PARTITION=$CHECK_SINGLE_PARTITION" > /dev/tty2
-				[ "$CHECK_SINGLE_PARTITION" == "no" ] && unset PARTITION
 				setup_install
 				;;
 			"$(gettext "Partitioning")")
@@ -2221,46 +2228,8 @@ setup_initrd() {
 				fi
 				;;
 			"$(gettext "Install")")
-				echo "CHECK_SINGLE_PARTITION=$CHECK_SINGLE_PARTITION (2256)" > /dev/tty2
-				[ "$CHECK_SINGLE_PARTITION" == "no" ] && unset PARTITION
 				setup_install
 				;;
-			"$(gettext "Partitioning")")
-				setup_partition
-				;;
-			"$(gettext "Format")")
-				echo "CHECK_SINGLE_PARTITION=$CHECK_SINGLE_PARTITION" > /dev/tty2
-				[ "$CHECK_SINGLE_PARTITION" == "no" ] && unset PARTITION
-				setup_format
-				;;
-			"$(gettext "Boot")")
-				echo "CHECK_SINGLE_PARTITION=$CHECK_SINGLE_PARTITION" > /dev/tty2
-				[ "$CHECK_SINGLE_PARTITION" == "no" ] && unset PARTITION
-				setup_grub
-				;;
-			"$(gettext "Keyboard")")
-				setup_keymap
-				;;
-			"$(gettext "Network")")
-				setup_net
-				;;
-			"$(gettext "Clock")")
-				setup_clock
-				;;
-			"$(gettext "Console")")
-				source /etc/profile
-				/usr/sbin/pwconv
-				/usr/sbin/grpconv
-				sh
-				;;
-			"$(gettext "Erase")")
-				echo "CHECK_SINGLE_PARTITION=$CHECK_SINGLE_PARTITION" > /dev/tty2
-				[ "$CHECK_SINGLE_PARTITION" == "no" ] && unset PARTITION
-				setup_erase
-				;;
-			"$(gettext "Restart")")
-				setup_reboot
-				;;
 		esac
 		;;
 	1)
@@ -2511,6 +2480,7 @@ top_loop() {
 			dialog --title "   $(gettext "Welcome to the NuTyX installer")   " \
 				--msgbox "    $(gettext "You can remove the ISO media now.")" 6 60
 		fi
+		check_available_disk
 		while [ "$SETUP_INSTALL" = "no" ];do
 				[ "$SETUP_CONFIG_LANGUAGE" = "yes" ] && LANG=de_CH.utf-8 setup_language
 				[ "$SETUP_INSTALL" = "no" ] && setup_initrd
