diff --git a/base/nutyx/Pkgfile b/base/nutyx/Pkgfile
index 5019912..2fb6952 100644
--- a/base/nutyx/Pkgfile
+++ b/base/nutyx/Pkgfile
@@ -8,11 +8,11 @@ maintainer="tnut <tnut@nutyx.org>"
 
 name=nutyx
 version=10.92.4
-# release=2
-# patchname="$name-$version-consolidate-$release.patch"
+release=2
+patchname="$name-$version-consolidate-$release.patch"
 
 source=(http://downloads.tuxfamily.org/nutyx/files/$name-$version.tar.xz)
-[ -z "$patchname ] || source+=" $patchname"
+[ -z "$patchname" ] || source+=" $patchname"
 
 build () {
 cd $name-$version
diff --git a/iso/isolinux/install.cfg b/iso/isolinux/install.cfg
index 0ec9bda..7ff0850 100755
--- a/iso/isolinux/install.cfg
+++ b/iso/isolinux/install.cfg
@@ -1,10 +1,10 @@
 default install
 label install
  menu label ^Install
- kernel /boot/kernel
- append initrd=/boot/initrd vga=788 ro quiet
+ kernel /boot/kernel-#VERSION_KERNEL#
+ append initrd=/boot/initrd-#VERSION_KERNEL# vga=788 ro quiet
 
 label install-debug
  menu label install-^Debug
- kernel /boot/kernel
- append initrd=/boot/initrd vga=788 ro testiso
+ kernel /boot/kernel-#VERSION_KERNEL#
+ append initrd=/boot/initrd-#VERSION_KERNEL#  vga=788 ro testiso
diff --git a/iso/isolinux/menu.cfg b/iso/isolinux/menu.cfg
index 146d20c..ec48474 100755
--- a/iso/isolinux/menu.cfg
+++ b/iso/isolinux/menu.cfg
@@ -1,7 +1,7 @@
 menu hshift 7
-menu width 65
+menu width 80
 
-menu title NuTyX GNU/Linux installer #ARCH# #VERSION_NUTYX# #VERSION_KERNEL#
+menu title NuTyX GNU/Linux installer #VERSION_NUTYX# #VERSION_KERNEL#
 include stdmenu.cfg
 include install.cfg
 include live.cfg
diff --git a/scripts/mkinitramfs.in b/scripts/mkinitramfs.in
index 234686f..2d0ba37 100755
--- a/scripts/mkinitramfs.in
+++ b/scripts/mkinitramfs.in
@@ -76,7 +76,7 @@ else
 	fi
 fi
 
-INITRAMFS_FILE=initrd-$KERNEL_VERSION
+INITRAMFS_FILE=initrd-$1
 
 if [ -n "$KERNEL_VERSION" ] && [ ! -d "/lib/modules/$1" ] ; then
   echo "No modules directory named $1"
@@ -356,8 +356,8 @@ mv $INITRAMFS_FILE /boot/$INITRAMFS_FILE
 
 if [ -d /ISO ];then
 	mkdir -p /ISO/boot
-	cp -v /boot/kernel-$KERNEL_VERSION /ISO/boot/kernel
-	cp -v /boot/initrd-$KERNEL_VERSION /ISO/boot/initrd
+	cp -v /boot/kernel-$1 /ISO/boot/
+	cp -v /boot/initrd-$1 /ISO/boot/
 fi
 
 printf "done.\n"
diff --git a/scripts/mkiso b/scripts/mkiso
index e5c4fc7..43f5596 100644
--- a/scripts/mkiso
+++ b/scripts/mkiso
@@ -74,6 +74,7 @@ done
 
 	ISO=NuTyX_${ARCH}-${VERSION_NUTYX}
 	sed -i "s|#VERSION_KERNEL#|${VERSION_KERNEL}|" isolinux/menu.cfg
+	sed -i "s|#VERSION_KERNEL#|${VERSION_KERNEL}|" isolinux/install.cfg
 	sed -i "s|#VERSION_NUTYX#|${VERSION_NUTYX}|" isolinux/menu.cfg
 	sed -i "s|#ARCH#|${ARCH}|" isolinux/menu.cfg
 
@@ -117,27 +118,29 @@ else
   set color_highlight=white/blue
 fi
 
-menuentry "NuTyX #ARCH# #VERSION_NUTYX# (UEFI mode) normal" {
+menuentry "NuTyX #ARCH# #VERSION_NUTYX# #VERSION_KERNEL# (UEFI mode) normal" {
 
-    linux /boot/kernel ro quiet
-    initrd /boot/initrd
+    linux /boot/kernel-#VERSION_KERNEL#  ro quiet
+    initrd /boot/initrd-#VERSION_KERNEL#
 }
-menuentry "NuTyX #ARCH# #VERSION_NUTYX# (UEFI mode) debug" {
+menuentry "NuTyX #ARCH# #VERSION_NUTYX# #VERSION_KERNEL# (UEFI mode) debug" {
     set background_color=black
-    linux /boot/kernel ro testiso rootdelay=3
-    initrd /boot/initrd
+    linux /boot/kernel-#VERSION_KERNEL#  ro testiso rootdelay=3
+    initrd /boot/initrd-#VERSION_KERNEL#
 }
-menuentry "NuTyX #ARCH# #VERSION_NUTYX# (UEFI mode) installable live" {
-    linux /boot/kernel ro quiet rootdelay=5 live
-    initrd /boot/initrd
+menuentry "NuTyX #ARCH# #VERSION_NUTYX# #VERSION_KERNEL# (UEFI mode) installable live" {
+    linux /boot/kernel-#VERSION_KERNEL#  ro quiet rootdelay=5 live
+    initrd /boot/initrd-#VERSION_KERNEL#
 }
-menuentry "NuTyX #ARCH# #VERSION_NUTYX# (UEFI mode) debug live" {
-    linux /boot/kernel ro testiso rootdelay=3 live
-    initrd /boot/initrd
+menuentry "NuTyX #ARCH# #VERSION_NUTYX# #VERSION_KERNEL# (UEFI mode) debug live" {
+    linux /boot/kernel-#VERSION_KERNEL#  ro testiso rootdelay=3 live
+    initrd /boot/initrd-#VERSION_KERNEL#
 }
 EOF
 	sed -i "s|#VERSION_NUTYX#|${VERSION_NUTYX}|" boot/grub/grub.cfg
+	sed -i "s|#VERSION_KERNEL#|${VERSION_KERNEL}|" boot/grub/grub.cfg
 	sed -i "s|#ARCH#|${ARCH}|" boot/grub/grub.cfg
+
 	for FILE in kernel initrd
 	do
 		[ -f isolinux/$FILE ] && mv -v isolinux/$FILE boot/
diff --git a/tools/lib/init.in b/tools/lib/init.in
index 27b306b..9bcc39c 100644
--- a/tools/lib/init.in
+++ b/tools/lib/init.in
@@ -195,7 +195,7 @@ do_mount_in_ram()
 		fi
 	done
 	if [ "$ROOTFS_OK" == "yes" ]; then
-		cp /media/cdrom/boot/kernel /boot/kernel
+		cp /media/cdrom/boot/kernel* /boot/
 		mkdir -p /boot/grub/fonts
 		cp /media/cdrom/boot/grub/fonts/unicode.pf2 /boot/grub/fonts/unicode.pf2
 		cp /media/cdrom/boot/grub/splash.png /boot/grub/splash.png
