diff --git a/scripts/setup-nutyx.in b/scripts/setup-nutyx.in
index 9e5a89d..7e28edd 100755
--- a/scripts/setup-nutyx.in
+++ b/scripts/setup-nutyx.in
@@ -1417,16 +1417,16 @@ check_available_partition() {
 			echo "device: $DISK" >> $TMP/PARTITIONNING
 			if [ -d "$EFIVARS" ];then
 				echo "
-${DISK}1 size=100M,type=uefi
-${DISK}2 size=4G,type=swap
-${DISK}3 type=linux" >> $TMP/PARTITIONNING
+${DISK}1: size=100M,type=uefi
+${DISK}2: size=4G,type=swap
+${DISK}3: type=linux" >> $TMP/PARTITIONNING
 			else
 				echo "
-${DISK}1 size=4G,type=swap
-${DISK}2 type=linux" >> $TMP/PARTITIONNING
+${DISK}1: size=4G,type=swap
+${DISK}2: type=linux" >> $TMP/PARTITIONNING
 			fi
-			CHECK_PART_OK="yes"
-			sfdisk ${DISK} < $TMP/PARTITIONNING > /dev/tty2 || CHECK_PART_OK="new"
+			CHECK_PART_OK="new"
+			sfdisk ${DISK} < $TMP/PARTITIONNING > /dev/tty2 || CHECK_PART_OK="no"
 		fi
 	fi
 	PARTITION_LIST=`for i in $DISK_LIST; do lsblk -p -l -n -o NAME $i| tail -n +2; done`
