commit 00a634f9440db4254d0895f29984dc7b5a38d33b
Author: tnut <tnut@nutyx.org>
Date:   Thu Nov 26 15:37:31 2020 +0100

    install-nutyx,
    need bzip2 as well in a minimal install

diff --git a/scripts/install-nutyx.in b/scripts/install-nutyx.in
index 5ef5eb9a4..e885a010b 100755
--- a/scripts/install-nutyx.in
+++ b/scripts/install-nutyx.in
@@ -1128,7 +1128,7 @@ check_root() {
 }
 check_variables() {
 
-CR="nutyx busybox curl gcc glibc \
+CR="nutyx busybox curl gcc bzip2 glibc \
 zlib lz4 xz zstd libarchive openssl cards"
 
 if [ -z "$CHROOT" ]; then
diff --git a/scripts/setup-nutyx.in b/scripts/setup-nutyx.in
index 187703897..d892e1996 100755
--- a/scripts/setup-nutyx.in
+++ b/scripts/setup-nutyx.in
@@ -385,11 +385,16 @@ setup_wireless()
 {
 	local TITLE WPA_FILE returncode WPA_TOOL ESSID KEY
 	WPA_TOOL="$(which wpa_passphrase)"
+	if [ -z "${WPA_TOOL}" ]; then
+		error "wpa-supplicant not installed"
+	fi
 	IWCONFIG="$(which iwconfig)"
-
-	if [ ! -z $IWCONFIG ];then
-		WIFI_NETWORK_CARD="$($IWCONFIG 2>&1|grep IEEE|cut -d " " -f1|tail -1)"
+	if [ -z "${IWCONFIG}" ]; then
+		error "wireless-tools not installed"
 	fi
+
+	WIFI_NETWORK_CARD="$($IWCONFIG 2>&1|grep IEEE|cut -d " " -f1|tail -1)"
+
 	if [ -z ${WIFI_NETWORK_CARD} ];then
 		network_card_found
 		[ -z ${NETWORK_CARD} ] && return
@@ -434,6 +439,11 @@ setup_wireless()
 		--colors \
 		--msgbox "\Zb\Z1$(gettext "Cannot create") $WPA_FILE\Zn" 0 0
 	fi
+	mkdir -p /etc/init/single
+  mkdir -p /etc/init/multi
+  ln -sv /lib/init/services/wpa-supplicant /etc/init/single/wpa-supplicant
+  ln -sv /lib/init/services/wpa-supplicant /etc/init/multi/wpa-supplicant
+
 	SETUP_CONFIG_WIFI="no"
 }
 setup_net() 
