diff --git a/tools/lib/init-functions b/tools/lib/init-functions
index 093ac4a65..39a732699 100644
--- a/tools/lib/init-functions
+++ b/tools/lib/init-functions
@@ -878,7 +878,8 @@ do_mount_virtualfs()
                 mount /run || failed=1
         fi
         mkdir -p /run/lock /run/shm
-        chmod 1777 /run/shm
+        chmod 1777 /run/shm /run/lock
+
         log_info_msg "Mounting virtual file systems: ${INFO}/run"
         if ! mountpoint /proc >/dev/null; then
                 log_info_msg2 " ${INFO}/proc"
diff --git a/tools/lib/init.in b/tools/lib/init.in
index a6ca8260e..c838b7b72 100644
--- a/tools/lib/init.in
+++ b/tools/lib/init.in
@@ -274,6 +274,26 @@ ListModules() {
   printf "Press <enter> to continue ...\n"
   read a
 }
+do_start_lvm() {
+if [ -f /etc/mdadm.conf ]; then
+
+  [ "$verbose" == "no" ] || printf "mdadm -As...\n"
+  /sbin/mdadm -As
+
+  [ "$verbose" == "no" ] || printf "mdadm --auto-detect...\n"
+  /sbin/mdadm --auto-detect
+fi
+
+### First we activate lvm if possible
+  if [ -f /sbin/lvm ]; then
+    [ "$verbose" == "no" ] || printf "/sbin/lvm vgchange --ignorelockingfailure -ay\n"
+    /sbin/lvm vgchange --ignorelockingfailure -ay
+    if [ $? = 0 ]; then
+       [ "$verbose" == "no" ] || printf "/sbin/lvm vgscan --ignorelockingfailure --mknodes\n"
+       /sbin/lvm vgscan --ignorelockingfailure --mknodes
+    fi
+  fi
+}
 # **********
 # Begin here
 # **********
@@ -320,27 +340,10 @@ done
 if [ -d /lib/modules ] && [ -n "$rootdelay" ] && [ -n "$root" ]; then
         [ "$verbose" == "no" ] || printf "Start udev...\n"
         do_start_udev
+	[ "$verbose" == "no" ] || printf "Start lvm ...\n"
+	do_start_lvm
 fi
 
-if [ -f /etc/mdadm.conf ]; then
-
-  [ "$verbose" == "no" ] || printf "mdadm -As...\n"
-  /sbin/mdadm -As
-
-  [ "$verbose" == "no" ] || printf "mdadm --auto-detect...\n"
-  /sbin/mdadm --auto-detect
-fi  
-
-### First we activate lvm if possible
-  if [ -f /sbin/lvm ]; then
-    [ "$verbose" == "no" ] || printf "/sbin/lvm vgchange --ignorelockingfailure -ay\n"
-    /sbin/lvm vgchange --ignorelockingfailure -ay
-    if [ $? = 0 ]; then
-       [ "$verbose" == "no" ] || printf "/sbin/lvm vgscan --ignorelockingfailure --mknodes\n"
-       /sbin/lvm vgscan --ignorelockingfailure --mknodes
-    fi
-  fi
-
 if [ -n "$rootdelay" ] ; then
 	[ "$verbose" == "no" ] || printf "Wait  %s second(s) ...\n" $rootdelay
 	sleep "$rootdelay"
@@ -417,6 +420,9 @@ if [ "$root" == "" ]; then
 	[ "$verbose" == "no" ] || printf "Start udev ...\n"
         do_start_udev
 
+	[ "$verbose" == "no" ] || printf "Start lvm ...\n"
+	do_start_lvm
+
 	[ "$verbose" == "no" ] || printf "Start sysklogd ...\n"
 	do_start_sysklogd
 	[ "$verbose" == "no" ]  && clear
