diff -Naur a/charproc.c b/charproc.c
--- a/charproc.c	2019-02-11 11:21:54.000000000 +0100
+++ b/charproc.c	2019-03-13 19:51:39.039992132 +0100
@@ -8624,11 +8624,11 @@
 #define MyWhitePixel(dpy) WhitePixel(dpy,DefaultScreen(dpy))
 
     if (request->misc.re_verse) {
-	wnew->dft_foreground = MyWhitePixel(screen->display);
-	wnew->dft_background = MyBlackPixel(screen->display);
-    } else {
 	wnew->dft_foreground = MyBlackPixel(screen->display);
 	wnew->dft_background = MyWhitePixel(screen->display);
+    } else {
+	wnew->dft_foreground = MyWhitePixel(screen->display);
+	wnew->dft_background = MyBlackPixel(screen->display);
     }
 
     init_Tres(TEXT_FG);
