diff -Naur a/charproc.c b/charproc.c
--- a/charproc.c	2019-07-22 21:14:11.000000000 +0000
+++ b/charproc.c	2019-08-18 20:14:50.529647845 +0000
@@ -8733,11 +8733,11 @@
 #define MyWhitePixel(dpy) WhitePixel(dpy,DefaultScreen(dpy))
 
     if (request->misc.re_verse) {
-	wnew->dft_foreground = MyWhitePixel(screen->display);
-	wnew->dft_background = MyBlackPixel(screen->display);
-    } else {
 	wnew->dft_foreground = MyBlackPixel(screen->display);
 	wnew->dft_background = MyWhitePixel(screen->display);
+    } else {
+	wnew->dft_foreground = MyWhitePixel(screen->display);
+	wnew->dft_background = MyBlackPixel(screen->display);
     }
 
     init_Tres(TEXT_FG);
