diff --git a/src/compile_dependencies_utils.cxx b/src/compile_dependencies_utils.cxx
index 67d6f01..7eb57b4 100644
--- a/src/compile_dependencies_utils.cxx
+++ b/src/compile_dependencies_utils.cxx
@@ -55,10 +55,20 @@ void addDepToDepList(depList *list, unsigned int nameIndex, int niveau)
 void freeDepList(depList *list)
 {
 	for (unsigned int i = 0 ; i < list->count;i++) {
-		free(list->niveau);
+		if  (list->niveau != nullptr) {
+			free(list->niveau);
+			list->niveau=nullptr;
+		}
 		free(list->depsIndex);
+		if  (list->depsIndex != nullptr) {
+			free(list->depsIndex);
+			list->depsIndex=nullptr;
+		}
+	}
+	if (list != nullptr) {
+		free(list);
+		list=nullptr;
 	}
-	free(list);
 }
 pkgInfo *initPkgInfo(void)
 {
@@ -81,7 +91,10 @@ pkgInfo *addInfoToPkgInfo(unsigned int nameIndex)
 
 void freePkgInfo(pkgInfo *package)
 {
-	free(package);
+	if (package !=nullptr) {
+		free(package);
+		package=nullptr;
+	}
 }
 
 pkgList *initPkgList(void)
@@ -112,10 +125,19 @@ void addPkgToPkgList(pkgList *list, pkgInfo *package)
 void freePkgList(pkgList *packagesList)
 {
 	for (unsigned int i = 0 ; i < packagesList->count;i++) {
-		free(packagesList->pkgs[i]);
+		if (packagesList->pkgs[i] != nullptr) {
+			free(packagesList->pkgs[i]);
+			packagesList->pkgs[i]=nullptr;
+		}
+	}
+	if (packagesList->pkgs != nullptr) {
+		free( packagesList->pkgs);
+		packagesList=nullptr;
+	}
+	if (packagesList != nullptr) {
+		free(packagesList);
+		packagesList=nullptr;
 	}
-	free( packagesList->pkgs);
-	free(packagesList);
 }
 
 /* Get the tree dependencies of the dependencies recursively */
@@ -127,7 +149,7 @@ int deps_tree (pkgList *packagesList, depList *dependenciesList,unsigned int nam
 	}
 	depList *localDependenciesList = initDepsList();
 	if ( dependenciesList->count > packagesList->count ) {
-		printf ("Dependencies List Bigger than Package List. \nDue to circular !!!\n");
+		printf ("Dependencies List Bigger than Package List: %d dependencies. \nDue to circular !!!\n",dependenciesList->count);
 		freePkgList(packagesList);
 		freeDepList(dependenciesList);
 		exit(EXIT_FAILURE);
