diff --git a/scripts/pkgmk.in b/scripts/pkgmk.in
index 10d5924..b2e3e1c 100644
--- a/scripts/pkgmk.in
+++ b/scripts/pkgmk.in
@@ -1027,7 +1027,7 @@ build_needed() {
 update_collection_repo() {
 	local HEAD BUILD_DATE VERSION RELEASE DESCRIPTION URL MAINTAINER PACKAGER EXTENSION ALIAS GROUP
 	if [ -f $PKGMK_REPO ]; then
-		for i in `tail -n +3  $PKGMK_REPO`
+		for i in `tail -n +2  $PKGMK_REPO`
 		do
 			if [ "`echo $i|cut -d "#" -f3`" != "" ];then
 				GROUP="`echo $i|cut -d "#" -f2|cut -d "." -f2` $GROUP"
