diff --git a/scripts/pkgmk.in b/scripts/pkgmk.in
index 68dba7b..6e2dc1a 100755
--- a/scripts/pkgmk.in
+++ b/scripts/pkgmk.in
@@ -741,6 +741,7 @@ add_meta_to_archive() {
 	echo "r$release" >> .META
 	echo "S$SIZE_I" >> .META
 	echo "c$COLLECTION" >> .META
+	echo "a$PKGMK_ARCH" >> .META
 
 	if [ -z "$GROUP" ]; then
 		echo "D$DESC" >> .META
@@ -749,7 +750,6 @@ add_meta_to_archive() {
 		echo "M$MAINT" >> .META
 		echo "P$PACK" >> .META
 		echo "L$LICENSE" >> .META
-		echo "a$PKGMK_ARCH" >> .META
 		echo "B$PKGMK_BUILDVER" >> .META
 	else
 	    echo "g$GROUP" >> .META
diff --git a/src/create.cxx b/src/create.cxx
index 6f4d4e1..1afda99 100644
--- a/src/create.cxx
+++ b/src/create.cxx
@@ -425,7 +425,7 @@ void create::buildBinary(std::string packageName)
             + " "
             + version;
         m_upgrade = 0;
-        buildDatabase(false, false);
+        buildDatabase(false, true);
 		if (checkPackageNameExist(name)) {
 			message = name + ": is ALLREADY installed";
 			m_upgrade = 1;
