diff --git a/scripts/build-port.in b/scripts/build-port.in
index f320c0e..5bb0226 100644
--- a/scripts/build-port.in
+++ b/scripts/build-port.in
@@ -107,7 +107,6 @@ check_ports()
 		echo "Variable 'PORT_BASE'"
 		exit 1
 	fi
-	sync_pkgmk_conf
 	pushd ${PORT_BASE} > /dev/null
 	for DIR in *
 	do
@@ -391,7 +390,6 @@ remove_url () {
 }
 build_binaries () {
 	local package port_version binary_version
-	sync_pkgmk_conf
 	if [ -n "$(cards level|grep WARNING|head -1)" ]; then
 		cards level
 		exit 1;
