diff --git a/scripts/pkgmk.in b/scripts/pkgmk.in
index 3a9e8e9..92f1bb9 100755
--- a/scripts/pkgmk.in
+++ b/scripts/pkgmk.in
@@ -769,6 +769,18 @@ build() {
 				make
 				make DESTDIR=$PKG install;;
 
+			kde6)
+				mkdir build
+				cd    build
+				cmake -DCMAKE_INSTALL_PREFIX=/usr  \
+					-DCMAKE_BUILD_TYPE=Release \
+					-DBUILD_TESTING=OFF        \
+					-Wno-dev \
+					../$name-$version
+				make
+				make DESTDIR=$PKG install;;
+
+
 			*)
 				echo "$(basename $PKGMK_COMMAND): function build invalid build type"
 				exit $E_BUILD ;;
