--- b/scripts/pkgmk.in	2024-03-05 21:34:28.000000000 +0100
+++ /home/thierry/git/nutyx/cards/scripts/pkgmk.in	2024-05-02 14:27:47.994504413 +0200
@@ -711,7 +711,8 @@
 			meson)
 				mkdir build
 				cd    build
-				meson --prefix=/usr \
+				meson setup --prefix=/usr \
+                                      --buildtype=release \
                                       ../$name-$version
 				ninja
 				DESTDIR=$PKG ninja install;;
@@ -769,6 +770,21 @@
 				make
 				make DESTDIR=$PKG install;;
 
+			kde6)
+				mkdir build
+				cd    build
+				cmake -DCMAKE_INSTALL_PREFIX=/usr  \
+					-DCMAKE_INSTALL_LIBEXECDIR=libexec \
+					-DCMAKE_BUILD_TYPE=Release \
+					-DCMAKE_INSTALL_LIBDIR=lib \
+					-DBUILD_QT5=OFF            \
+					-DBUILD_TESTING=OFF        \
+					-Wno-dev \
+					../$name-$version
+				make
+				make DESTDIR=$PKG install;;
+
+
 			*)
 				echo "$(basename $PKGMK_COMMAND): function build invalid build type"
 				exit $E_BUILD ;;
@@ -1802,12 +1818,13 @@
 
 readonly PKGMK_VERSION="#VERSION#"
 readonly PKGMK_COMMAND="$0"
-readonly PKGMK_ROOT="$PWD"
+PKGMK_ROOT="$PWD"
 
 PKGMK_PACKAGE_DIR="$PWD"
 PKGMK_BUILDVER="$(date +%s)"
 PKGMK_PACKAGE_EXT="cards.tar"
 PKGMK_REPO=".PKGREPO"
+PKGMK_REPO_FILE=".REPO"
 PKGMK_ARCH="$(uname -m)"
 
 PKGMK_CONFFILE="/etc/pkgmk.conf"
