diff --git a/scripts/mybld.in b/scripts/mybld.in
old mode 100644
new mode 100755
index ff659af..5c51bd3
--- a/scripts/mybld.in
+++ b/scripts/mybld.in
@@ -220,6 +220,10 @@ case $1 in
 		shift
 		PARAM+=($1);;
 
+	-p|--pause)
+		PAUSE=" && export PATH=/root/$VERSION/scripts:$PATH && printf \"You are now in the chroot.\n\
+Type exit when finish.\n\n Good Luck !!!\n\n\" && bash";;
+
 	-I|--init)
 		INIT="yes";;
 
@@ -456,9 +460,11 @@ if [ ! -d .git ]; then
 "
 	exit 1
 fi
-parse_arguments "$@"
 
 [ -z "$VERSION" ] && VERSION="$(git branch --show-current)"
+
+parse_arguments "$@"
+
 [ -z "$COLLECTION" ] && find_collection
 
 if [ "$INIT" == "yes" ]; then
@@ -552,6 +558,12 @@ GIT=$(basename $GIT_PROJ)
 mkdir -p "$ROOT/${VERSION}/${COLLECTION}/etc" || exit 1
 
 setup_cards_conf
+
+DEVEL="cards.devel \
+&& cd /root/$GIT \
+$PAUSE \
+&& bash scripts/$COLLECTION"
+
 echo "$CARDS_CONF" > $ROOT/$VERSION/$COLLECTION/etc/cards.conf
 [ -z "${PARAM[0]}" ] && CHROOT="ncurses readline bash"
 if [ "$COLLECTION" == "base" ] \
@@ -561,18 +573,13 @@ if [ "$COLLECTION" == "base" ] \
 || [ "$COLLECTION" == "gui-extra" ] \
 || [ "$COLLECTION" == "desktops" ] \
 || [ "$COLLECTION" == "nos" ]; then
-	DEVEL="cards.devel \
-&& cd /root/$GIT \
-&& bash scripts/$COLLECTION"
 	if ! [ -z "${PARAM[0]}" ]; then
 		DEVEL="$DEVEL ${PARAM[0]}"
 	else
 		DEVEL="$DEVEL -a"
 	fi
 else
-	DEVEL="cards.devel \
-&& cd /root/$GIT \
-&& bash scripts/$COLLECTION -a \
+	DEVEL="$DEVEL -a \
 && bash scripts/$COLLECTION-extra -a"
 	EXTRA="yes"
 fi
diff --git a/scripts/pkgmk.in b/scripts/pkgmk.in
index d169bd7..1b3a17b 100644
--- a/scripts/pkgmk.in
+++ b/scripts/pkgmk.in
@@ -327,7 +327,7 @@ unpack_source() {
 	for FILE in ${source[@]}; do
 		LOCAL_FILENAME="$(get_filename $FILE)"
 		case $LOCAL_FILENAME in
-			*.tar|*.tar.gz|*.tar.Z|*.tgz|*.tar.bz2|*.tbz2|*.tar.xz|*.txz|*.tar.lzma|*.zip|*.rpm|*.tar.lz|*.7z|*.deb)
+			*.tar|*.tar.gz|*.tar.Z|*.tgz|*.tar.bz2|*.tbz2|*.tar.xz|*.txz|*.tar.lzma|*.zip|*.tar.zst|*.rpm|*.tar.lz|*.7z|*.deb)
 				if [ "$PKGMK_IGNORE_UNPACK" != "yes" ];then
 					COMMAND="bsdtar -p -o -C $SRC -xf $LOCAL_FILENAME"
 				else
diff --git a/src/cards_install.cxx b/src/cards_install.cxx
index fad833b..b017518 100644
--- a/src/cards_install.cxx
+++ b/src/cards_install.cxx
@@ -46,7 +46,7 @@ Cards_install::Cards_install(const CardsArgumentParser& argParser,
 		std::set<std::string> ListOfPackage = getListOfPackagesFromSet(i);
 		if (ListOfPackage.empty() )
 			ListOfPackage = getListOfPackagesFromCollection(i);
-		if ( (!ListOfPackage.empty()) && (!checkBinaryExist(i)) ) {
+		if ( !ListOfPackage.empty() )  {
 			/*
 			* It's a collection or a set
 			*/
