diff --git a/src/create.cxx b/src/create.cxx
index 8dd5438..11d5c70 100644
--- a/src/create.cxx
+++ b/src/create.cxx
@@ -9,6 +9,12 @@ create::create(CardsArgumentParser& argParser)
     , m_config("/etc/cards.conf")
     , m_fdlog(-1)
 {
+    char pwd[MAXPATHLEN];
+    m_portsDir = getcwd(pwd,MAXPATHLEN);
+    std::cout << "Current Directory: "
+        << m_portsDir
+        << std::endl;
+
     parseArguments();
     m_tree = m_pkgfile.getListOfPackages();
 
@@ -141,11 +147,15 @@ void create::build(std::string packageName)
     installDependencies(packageName);
 
     std::cout << "create of " << packageName << std::endl;
-    std::string pkgdir = m_pkgfile.getPortDir(packageName);
-    if (pkgdir.empty()) {
+
+    if (m_pkgfile.getPortDir(packageName).empty()) {
         m_actualError = cards::ERROR_ENUM_PACKAGE_NOT_FOUND;
         treatErrors(packageName);
     }
+
+    m_portsDir += "/";
+    m_portsDir += m_pkgfile.getPortDir(packageName);
+
     std::string timestamp, message;
     std::string commandName = "cards create: ";
 
@@ -165,7 +175,6 @@ void create::build(std::string packageName)
     std::cout << "Current Directory: "
         << getcwd(pwd,100)
         << std::endl;
-
     if (chdir(pkgdir.c_str())) {
         m_actualError = cards::ERROR_ENUM_CANNOT_CHANGE_DIRECTORY;
         treatErrors(pkgdir);
