diff --git a/man/pkgmk.conf.5.fr.in b/man/pkgmk.conf.5.fr.in
index 887665c..9cfc6b1 100644
--- a/man/pkgmk.conf.5.fr.in
+++ b/man/pkgmk.conf.5.fr.in
@@ -107,7 +107,7 @@ Défaut: 'yes'
 Si défini, pkgmk cherchera dans les dossiers 'STRING1', .., 'STRINGN' pour y trouver d'eventuelles dépendances. Chaque dossier étant défini dans cette variable.
 .br
 Default: '()' pkgmk trouvera les dépendances de manière automatique.
-Possible values: '(base cli cli-extra gui gui-extra desktops <votre_collection>)'
+Possible values: '(base cli cli-extra gui gui-extra <votre_collection>)'
 .TP
 \fBPKGMK_GROUPS=('STRING1' .. 'STRINGN')
 Si défini, pkgmk fera une tentative de séparation du paquet produit en sous groupes. Chaque groupe étant défini dans cette variable. Une fonction par défaut est exécutée si cette même fonction n'est pas trouvée dans le Pkgfile. Les groupes de fonction suivantes sont disponibles: lib() devel()  doc()  man()  service()
diff --git a/man/pkgmk.conf.5.in b/man/pkgmk.conf.5.in
index 1c73473..3fb1dc7 100644
--- a/man/pkgmk.conf.5.in
+++ b/man/pkgmk.conf.5.in
@@ -112,7 +112,7 @@ Default: 'yes'
 If set, pkgmk will look into 'STRING1', .., 'STRINGN' directories for any dependencies. Each directory is define in this variable.
 .br
 Default: '()' pkgmk will find them automatically.
-Possible values: '(base cli cli-extra gui gui-extra desktops <your_collection>)'
+Possible values: '(base cli cli-extra gui gui-extra <your_collection>)'
 .TP
 \fBPKGMK_GROUPS=('STRING1' .. 'STRINGN')
 If set, pkgmk will try to split the produced package into groups. Each group is define in this variable. A default function is executed if this same function is not found in the Pkgfile. Supported default group functions are: lib() devel()  doc()  man()  service()
diff --git a/scripts/mybld.in b/scripts/mybld.in
index adb5cfa..ea20f1f 100755
--- a/scripts/mybld.in
+++ b/scripts/mybld.in
@@ -348,8 +348,8 @@ CARDS_CONF=""
 C="${COLLECTION}"
 
 [[ "${COLLECTION}" != "base" && "${COLLECTION}" != "cli" && "${COLLECTION}" != "gui" && \
-"${COLLECTION}" != "nos" && "${COLLECTION}" != "cli-extra" && "${COLLECTION}" != "gui-extra" && \
-"${COLLECTION}" != "desktops" ]] && C="${COLLECTION}-extra"
+"${COLLECTION}" != "nos" && "${COLLECTION}" != "cli-extra" && "${COLLECTION}" != "gui-extra"  ]] \
+&& C="${COLLECTION}-extra"
 
 if [ "${C}" == "${COLLECTION}-extra" ]; then
 	CARDS_CONF+="
@@ -375,15 +375,8 @@ dir /DEPOT/nos"
 base /DEPOT/nos"
 fi
 
-if [ "${C}" == "desktops" ]; then
-	CARDS_CONF+="
-dir /DEPOT/desktops"
-	[ -z "$BASE" ] && CARDS_CONF+="
-base /DEPOT/desktops"
-fi
-
 if [[ "${C}" == "gui" || "${C}" == "gui-extra" || \
-"${C}" == "nos" || "${C}" == "desktops" ]]; then
+"${C}" == "nos" ]]; then
 	CARDS_CONF+="
 dir /DEPOT/gui"
 	[ -z "$BASE" ] && CARDS_CONF+="
@@ -398,7 +391,7 @@ base /DEPOT/cli-extra"
 fi
 
 if [[ "${C}" == "cli" || "${C}" == "cli-extra" || "${C}" == "gui" || \
-"${C}" == "gui-extra" || "${C}" == "nos" || "${C}" == "desktops" ]]; then
+"${C}" == "gui-extra" || "${C}" == "nos" ]]; then
 	CARDS_CONF+="
 dir /DEPOT/cli"
 	[ -z "$BASE" ] && CARDS_CONF+="
@@ -571,7 +564,6 @@ if [ "$COLLECTION" == "base" ] \
 || [ "$COLLECTION" == "gui" ] \
 || [ "$COLLECTION" == "cli-extra"  ] \
 || [ "$COLLECTION" == "gui-extra" ] \
-|| [ "$COLLECTION" == "desktops" ] \
 || [ "$COLLECTION" == "nos" ]; then
 	if ! [ -z "${PARAM[0]}" ]; then
 		DEVEL="$DEVEL ${PARAM[0]}"
diff --git a/src/pkgdbh.cxx b/src/pkgdbh.cxx
index 8fde4ab..a3265ae 100644
--- a/src/pkgdbh.cxx
+++ b/src/pkgdbh.cxx
@@ -285,7 +285,7 @@ int Pkgdbh::getNumberOfPackages()
   return getListOfPackageNames("");
 }
 /* Append to the "DB" the number of packages founds
- * (directory containg a file named files
+ * (directory containing a file named files
  * */
 int Pkgdbh::getListOfPackageNames (const std::string& path)
 {
@@ -307,7 +307,7 @@ int Pkgdbh::getListOfPackageNames (const std::string& path)
 #endif
 	return m_packageNamesList.size();
 }
-/* get details infos of a package */
+/* get details info of a package */
 std::pair<std::string, pkginfo_t> Pkgdbh::getInfosPackage
 	(const std::string& packageName)
 {
