diff --git a/scripts/mybld.in b/scripts/mybld.in
index d3a5364..8a52d80 100755
--- a/scripts/mybld.in
+++ b/scripts/mybld.in
@@ -426,7 +426,7 @@ setup_pkgmk_conf() {
 [ -z  "$PKGMK_WORK_DIR" ] && PKGMK_WORK_DIR="/tmp/work"
 [ -z  "$PKGMK_COMPRESSION_OPTS" ] && PKGMK_COMPRESSION_OPTS="-9 --threads=$(getconf _NPROCESSORS_ONLN)"
 [ -z  "$PKGMK_COMPRESS_PACKAGE" ] && PKGMK_COMPRESS_PACKAGE="yes"
-[ -z  "$PKGMK_COMPRESSION_MODE" ] && PKGMK_COMPRESSION_MODE="xz"
+[ -z  "$PKGMK_COMPRESSION_MODE" ] && PKGMK_COMPRESSION_MODE="zst"
 [ -z  "$PKGMK_LOCALES" ] && PKGMK_LOCALES="af an ar as az be bg \
 bs ca cs cy da de el es fi fr ga gl gn he hi hr hu ia id is it \
 ja ka kk km kn ko lt lv mk ml mr ms my nl nn oc or pa pt pl rm \
diff --git a/scripts/pkgmk.in b/scripts/pkgmk.in
index edbde0c..54f17dd 100755
--- a/scripts/pkgmk.in
+++ b/scripts/pkgmk.in
@@ -314,10 +314,11 @@ unpack_source() {
 	done
 }
 get_packages_list() {
-	local i FILE
+	local i FILE FILTER
+	FILTER=${PKGMK_PACKAGE_EXT##*.}
 	i=0
 	unset TARGETS
-	for FILE in $(find $PKGMK_PACKAGE_DIR -name "$name*$PKGMK_PACKAGE_EXT*"); do
+	for FILE in $(find $PKGMK_PACKAGE_DIR -name "$name.*${FILTER}*"); do
 		TARGETS[i]="$(basename $FILE)"
 		i=$((i+1))
 	done
@@ -401,11 +402,9 @@ check_footprint() {
 	local TARGET FILE="$PKGMK_WORK_DIR/.tmp"
 
 	cd $PKGMK_ROOT
+	get_packages_list
 	if [ -z $TARGETS ]; then
-		get_packages_list
-		if [ -z $TARGETS ]; then
-			error "Unable to update footprint"
-		fi
+		error "Unable to update footprint"
 	fi
 	for TARGET in ${TARGETS[@]}; do
 		if [ -f $PKGMK_PACKAGE_DIR/$TARGET ]; then
