diff --git a/src/cards_install.cpp b/src/cards_install.cpp
index 58b88a5..164b105 100644
--- a/src/cards_install.cpp
+++ b/src/cards_install.cpp
@@ -91,16 +91,14 @@ Cards_install::Cards_install(const CardsArgumentParser& argParser,
 				continue;
 			m_packageArchiveName = i + "/" + j;
 			ArchiveUtils packageArchive(m_packageArchiveName.c_str());
-			std::string name = "(" + packageArchive.collection()+") ";
-			name += packageArchive.name();
+			std::string name = packageArchive.name();
 			if ( ! checkPackageNameExist(name )) {
+				name = "(" + packageArchive.collection()+") " + name;
 				m_upgrade=0;
-			} else {
-				m_upgrade=1;
+				m_force=0;
+				run();
+				syslog(LOG_INFO,name.c_str());
 			}
-			m_force=0;
-			run();
-			syslog(LOG_INFO,name.c_str());
 		}
 	}
 }
