diff --git a/scripts/pkgmk.in b/scripts/pkgmk.in
index f8b17ea..67c1c86 100644
--- a/scripts/pkgmk.in
+++ b/scripts/pkgmk.in
@@ -1258,6 +1258,9 @@ check_sources_uptodate() {
 }
 check_build() {
 	local LIST
+	if [ "$PKGMK_KEEP_LA_LIBS" != "yes" ]; then
+		find $PKG ! -type d -name "*.la" -exec rm -fv -- '{}' +
+	fi
 	LIST="$(find $PKG ! -type d -name "*.la")"
 	if [ ! -z "$LIST" ]; then
 		warning "following files should be removed:"
@@ -1647,9 +1650,6 @@ main() {
 			build_package $TARGET
 		fi
 	fi
-	if [ "$PKGMK_KEEP_LA_LIBS" != "yes" ]; then
-		find $PKG ! -type d -name "*.la" -exec rm -fv -- '{}' +
-	fi
 	if [ "$PKGMK_INSTALL" != "no" ]; then
 		install_package
 	fi
