diff -Naur a/scripts/libcards.pc.in b/scripts/libcards.pc.in
--- a/scripts/libcards.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ b/scripts/libcards.pc.in	2023-06-05 14:17:16.038619777 +0200
@@ -0,0 +1,12 @@
+prefix=/usr
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: libcards
+URL: https://github.com/NuTyX/cards/
+Description: Library for cards package manager
+Version: #VERSION#
+Libs: -L${libdir} -lcards
+Libs.private:  -lcards
+Cflags: -I${includedir}
