diff --git a/scripts/mybld.in b/scripts/mybld.in
index 5c51bd3..b2335fb 100755
--- a/scripts/mybld.in
+++ b/scripts/mybld.in
@@ -221,7 +221,7 @@ case $1 in
 		PARAM+=($1);;
 
 	-p|--pause)
-		PAUSE=" && export PATH=/root/$VERSION/scripts:$PATH && printf \"You are now in the chroot.\n\
+		PAUSE=" && export PATH=/root/$GIT/scripts:$PATH && printf \"You are now in the chroot.\n\
 Type exit when finish.\n\n Good Luck !!!\n\n\" && bash";;
 
 	-I|--init)
@@ -463,6 +463,15 @@ fi
 
 [ -z "$VERSION" ] && VERSION="$(git branch --show-current)"
 
+GIT_PROJ=$(pwd)
+
+if [ -z "$GIT_PROJ" ]; then
+	error "Git root: ${GIT_PROJ} empty"
+	exit 1
+fi
+
+GIT=$(basename $GIT_PROJ)
+
 parse_arguments "$@"
 
 [ -z "$COLLECTION" ] && find_collection
@@ -546,15 +555,6 @@ if ! [ -O "$ROOT/$VERSION" ]; then
 	exit 1
 fi
 
-GIT_PROJ=$(pwd)
-
-if [ -z "$GIT_PROJ" ]; then
-	error "Git root: ${GIT_PROJ} empty"
-	exit 1
-fi
-
-GIT=$(basename $GIT_PROJ)
-
 mkdir -p "$ROOT/${VERSION}/${COLLECTION}/etc" || exit 1
 
 setup_cards_conf
