diff --git a/.gitignore b/.gitignore
index a0e9a31..cfcb0af 100644
--- a/.gitignore
+++ b/.gitignore
@@ -25,3 +25,4 @@ documentation/latex
 *.mo
 *.save
 *.layout
+build/
diff --git a/src/pkginfo.cxx b/src/pkginfo.cxx
index 7a19b10..98f364d 100644
--- a/src/pkginfo.cxx
+++ b/src/pkginfo.cxx
@@ -171,9 +171,9 @@ void Pkginfo::run()
 			buildSimpleDatabase();
 			for (auto i : m_listOfInstPackages) {
 				if ( i.second.dependency == true)
-					cout << "AUTO ";
+					cout << "AUTO: ";
 				else
-					cout << " MAN ";
+					cout << " MAN: ";
 
 				cout << "(" << i.second.collection << ")"
 				<< " " << i.first << " "
