diff --git a/scripts/mybld.in b/scripts/mybld.in
index 163b937..eba222b 100755
--- a/scripts/mybld.in
+++ b/scripts/mybld.in
@@ -358,7 +358,7 @@ base /DEPOT/$1"
 setup_cards_conf() {
   CARDS_CONF=""
   FOUND="no"
-  MYCOLLECTION="${COLLECTION%-*}"
+  MYCOLLECTION="${COLLECTION%-extra}"
   for i in "${COLLECTIONS[@]}"
   do
      [ $i == $MYCOLLECTION ] && FOUND="yes"
diff --git a/scripts/build-port.in b/scripts/build-port.in
index d03d51a..7485b57 100644
--- a/scripts/build-port.in
+++ b/scripts/build-port.in
@@ -473,3 +473,7 @@ GIT="$(pwd)"
 PROJECT="$(basename $GIT)"
 PKGMK_CONFFILE=".pkgmk.conf"
 main "$@"
+[ -f /etc/pkgmk.conf ] && . /etc/pkgmk.conf
+[ ! ${PKGMK_WORK_DIR} ] && exit 0
+cd ${PKGMK_WORK_DIR}
+bash
