diff -Naur a/gdk/gdkpixbuf-drawable.c b/gdk/gdkpixbuf-drawable.c
--- a/gdk/gdkpixbuf-drawable.c	2022-08-23 16:32:47.000000000 +0200
+++ b/gdk/gdkpixbuf-drawable.c	2022-11-26 16:00:57.880052814 +0100
@@ -104,7 +104,7 @@
    * by external applications.
    * So be on the safe side and:
    */
-  cairo_surface_mark_dirty (surface);
+  cairo_surface_flush (surface);
 
   if (cairo_surface_get_content (surface) & CAIRO_CONTENT_ALPHA)
     copy = cairo_image_surface_create (CAIRO_FORMAT_ARGB32, width * scale, height * scale);
diff -Naur a/gdk/wayland/cursor/meson.build b/gdk/wayland/cursor/meson.build
--- a/gdk/wayland/cursor/meson.build	1970-01-01 01:00:00.000000000 +0100
+++ b/gdk/wayland/cursor/meson.build	2022-11-26 15:59:48.213383271 +0100
@@ -0,0 +1,12 @@
+wayland_cursor_sources = files([
+  'wayland-cursor.c',
+  'xcursor.c',
+  'os-compatibility.c'
+])
+
+libwayland_cursor = static_library('wayland+cursor',
+  sources: wayland_cursor_sources,
+  include_directories: [ confinc, ],
+  dependencies: [ glib_dep, wlclientdep, ],
+  c_args: common_cflags,
+)
diff -Naur a/gdk/wayland/gdkdisplay-wayland.c b/gdk/wayland/gdkdisplay-wayland.c
--- a/gdk/wayland/gdkdisplay-wayland.c	2022-11-22 18:58:44.000000000 +0100
+++ b/gdk/wayland/gdkdisplay-wayland.c	2022-11-26 16:00:59.650052903 +0100
@@ -1113,6 +1113,26 @@
 }
 
 static struct wl_cursor_theme *
+try_load_theme (GdkWaylandDisplay *display_wayland,
+                const char        *dir,
+                gboolean           dotdir,
+                const char        *name,
+                int                size)
+{
+  struct wl_cursor_theme *theme = NULL;
+  char *path;
+
+  path = g_build_filename (dir, dotdir ? ".icons" : "icons", name, "cursors", NULL);
+
+  if (g_file_test (path, G_FILE_TEST_IS_DIR))
+    theme = wl_cursor_theme_create (path, size, display_wayland->shm);
+
+  g_free (path);
+
+  return theme;
+}
+
+static struct wl_cursor_theme *
 get_cursor_theme (GdkWaylandDisplay *display_wayland,
                   const char *name,
                   int size)
@@ -1121,16 +1141,18 @@
   struct wl_cursor_theme *theme = NULL;
   int i;
 
+  theme = try_load_theme (display_wayland, g_get_user_data_dir (), FALSE, name, size);
+  if (theme)
+    return theme;
+
+  theme = try_load_theme (display_wayland, g_get_home_dir (), TRUE, name, size);
+  if (theme)
+    return theme;
+
   xdg_data_dirs = g_get_system_data_dirs ();
   for (i = 0; xdg_data_dirs[i]; i++)
     {
-      char *path = g_build_filename (xdg_data_dirs[i], "icons", name, "cursors", NULL);
-
-      if (g_file_test (path, G_FILE_TEST_IS_DIR))
-        theme = wl_cursor_theme_create (path, size, display_wayland->shm);
-
-      g_free (path);
-
+      theme = try_load_theme (display_wayland, xdg_data_dirs[i], FALSE, name, size);
       if (theme)
         return theme;
     }
