diff -Naur a/runit-2.1.2/etc/debian/ctrlaltdel b/runit-2.1.2/etc/debian/ctrlaltdel
--- a/runit-2.1.2/etc/debian/ctrlaltdel	2014-08-10 20:22:34.000000000 +0200
+++ b/runit-2.1.2/etc/debian/ctrlaltdel	2020-03-19 00:00:39.566666092 +0100
@@ -1,9 +1,9 @@
 #!/bin/sh
 
 PATH=/bin:/usr/bin
-MSG="System is going down in 14 seconds..."
+MSG="System is going down in 3 seconds..."
 
 # echo 'disabled.' ; exit
-touch /etc/runit/stopit
-chmod 100 /etc/runit/stopit && echo "$MSG" | wall
-/bin/sleep 14
+touch /lib/init/stopit
+chmod 100 /lib/init/stopit && echo "$MSG" | wall
+/bin/sleep 3
diff -Naur a/runit-2.1.2/src/runit.c b/runit-2.1.2/src/runit.c
--- a/runit-2.1.2/src/runit.c	2014-08-10 20:22:35.000000000 +0200
+++ b/runit-2.1.2/src/runit.c	2020-03-18 23:43:48.976623509 +0100
@@ -17,15 +17,14 @@
 #include "reboot_system.h"
 
 /* #define DEBUG */
-
-#define INFO "- runit: "
-#define WARNING "- runit: warning: "
-#define FATAL "- runit: fatal: "
+#define INFO "\033[1;36m [INFO] \033[1;0m"
+#define WARNING "\033[1;33m [WARNING] \033[1;0m"
+#define FATAL "\033[1;31m [ERROR] \033[1;0m"
 
 const char * const stage[3] ={
-  "/etc/runit/1",
-  "/etc/runit/2",
-  "/etc/runit/3" };
+  "/lib/init/1",
+  "/lib/init/2",
+  "/lib/init/3" };
 
 int selfpipe[2];
 int sigc =0;
@@ -90,7 +89,7 @@
   if (RB_DISABLE_CAD == 0) reboot_system(0);
 #endif
 
-  strerr_warn3(INFO, "$Id: 25da3b86f7bed4038b8a039d2f8e8c9bbcf0822b $",
+  strerr_warn3(INFO, "NuTyX",
                ": booting.", 0);
 
   /* runit */
diff -Naur a/runit-2.1.2/src/runit.dist b/runit-2.1.2/src/runit.dist
--- a/runit-2.1.2/src/runit.dist	2014-08-10 20:22:35.000000000 +0200
+++ b/runit-2.1.2/src/runit.dist	2020-03-18 23:37:21.833291990 +0100
@@ -1,2 +1,2 @@
-- runit: fatal: must be run as process no 1.
+^[[1;31m [ERROR] ^[[1;0mmust be run as process no 1.
 111
diff -Naur a/runit-2.1.2/src/runit.h b/runit-2.1.2/src/runit.h
--- a/runit-2.1.2/src/runit.h	2014-08-10 20:22:35.000000000 +0200
+++ b/runit-2.1.2/src/runit.h	2020-03-18 22:03:17.419985116 +0100
@@ -1,4 +1,4 @@
 #define RUNIT "/sbin/runit"
-#define STOPIT "/etc/runit/stopit"
-#define REBOOT "/etc/runit/reboot"
-#define CTRLALTDEL "/etc/runit/ctrlaltdel"
+#define STOPIT "/lib/init/stopit"
+#define REBOOT "/lib/init/reboot"
+#define CTRLALTDEL "/lib/init/ctrlaltdel"
diff -Naur a/runit-2.1.2/src/sv.c b/runit-2.1.2/src/sv.c
--- a/runit-2.1.2/src/sv.c	2014-08-10 20:22:34.000000000 +0200
+++ b/runit-2.1.2/src/sv.c	2020-03-18 21:16:13.329998348 +0100
@@ -32,7 +32,7 @@
 char *progname;
 char *action;
 char *acts;
-char *varservice ="/service/";
+char *varservice ="/etc/init/default/";
 char **service;
 char **servicex;
 unsigned int services;
