diff -Naur lxdm-0.5.3/data/themes/Industrial/greeter.ui lxdm/data/themes/Industrial/greeter.ui
--- lxdm-0.5.3/data/themes/Industrial/greeter.ui	2013-03-27 13:11:40.000000000 +0100
+++ lxdm/data/themes/Industrial/greeter.ui	2017-12-16 19:02:01.383075700 +0100
@@ -34,7 +34,7 @@
                   <object class="GtkImage" id="image1">
                     <property name="visible">True</property>
                     <property name="pixbuf">login.png</property>
-                    <property name="yalign">0.8</property>
+                    <property name="yalign">1</property>
                   </object>
                   <packing>
                     <property name="position">0</property>
diff -Naur lxdm-0.5.3/data/themes/Industrial/gtkrc lxdm/data/themes/Industrial/gtkrc
--- lxdm-0.5.3/data/themes/Industrial/gtkrc	2012-05-29 13:31:02.000000000 +0200
+++ lxdm/data/themes/Industrial/gtkrc	2017-12-16 19:45:06.451895878 +0100
@@ -7,7 +7,7 @@
 			function = FLAT_BOX
 			shadow=NONE
 			state = NORMAL
-			file="wave.svg"
+			file="nutyx.jpg"
 			stretch=TRUE
 			detail = "base"
 		}
diff -Naur lxdm-0.5.3/src/lxdm.c lxdm/src/lxdm.c
--- lxdm-0.5.3/src/lxdm.c	2015-11-23 12:15:52.000000000 +0100
+++ lxdm/src/lxdm.c	2017-12-16 17:20:17.146685157 +0100
@@ -1184,12 +1184,16 @@
 	{
 		if(!strcasecmp(name,"LXDE"))
 			exec = g_strdup("startlxde");
-		else if( !strcasecmp(name, "GNOME") )
-			exec = g_strdup("gnome-session");
+		else if( !strcasecmp(name, "MATE") )
+			exec = g_strdup("mate-session");
 		else if( !strcasecmp(name, "KDE") )
 			exec = g_strdup("startkde");
+		else if( !strcasecmp(name, "LXQT") )
+			exec = g_strdup("startlxqt");
 		else if( !strcasecmp(name, "XFCE") || !strcasecmp(name, "xfce4"))
 			exec = g_strdup("startxfce4");
+		else if( !strcasecmp(name, "TWM"))
+			exec = g_strdup("startx");
 		else
 			exec=g_strdup(name);
 	}
