diff -Naur a/clients/tests/test-client.py b/clients/tests/test-client.py
--- a/clients/tests/test-client.py	2020-08-18 23:05:13.000000000 +0200
+++ b/clients/tests/test-client.py	2020-09-28 14:46:04.393212851 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 from __future__ import print_function
 
diff -Naur a/examples/python/dbus/add-connection-compat.py b/examples/python/dbus/add-connection-compat.py
--- a/examples/python/dbus/add-connection-compat.py	2020-08-19 09:39:11.000000000 +0200
+++ b/examples/python/dbus/add-connection-compat.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2010 - 2012 Red Hat, Inc.
diff -Naur a/examples/python/dbus/add-connection.py b/examples/python/dbus/add-connection.py
--- a/examples/python/dbus/add-connection.py	2020-08-19 09:39:11.000000000 +0200
+++ b/examples/python/dbus/add-connection.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2010 - 2012 Red Hat, Inc.
diff -Naur a/examples/python/dbus/add-wifi-eap-connection.py b/examples/python/dbus/add-wifi-eap-connection.py
--- a/examples/python/dbus/add-wifi-eap-connection.py	2020-08-19 09:39:11.000000000 +0200
+++ b/examples/python/dbus/add-wifi-eap-connection.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2011 Red Hat, Inc.
diff -Naur a/examples/python/dbus/add-wifi-psk-connection.py b/examples/python/dbus/add-wifi-psk-connection.py
--- a/examples/python/dbus/add-wifi-psk-connection.py	2020-08-19 09:39:11.000000000 +0200
+++ b/examples/python/dbus/add-wifi-psk-connection.py	2020-09-28 14:46:04.386546184 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2015 Red Hat, Inc.
diff -Naur a/examples/python/dbus/create-bond.py b/examples/python/dbus/create-bond.py
--- a/examples/python/dbus/create-bond.py	2020-08-19 09:39:11.000000000 +0200
+++ b/examples/python/dbus/create-bond.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2015 Red Hat, Inc.
diff -Naur a/examples/python/dbus/disconnect-device.py b/examples/python/dbus/disconnect-device.py
--- a/examples/python/dbus/disconnect-device.py	2020-08-18 23:05:13.000000000 +0200
+++ b/examples/python/dbus/disconnect-device.py	2020-09-28 14:46:04.386546184 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2010 Red Hat, Inc.
diff -Naur a/examples/python/dbus/get-active-connection-uuids.py b/examples/python/dbus/get-active-connection-uuids.py
--- a/examples/python/dbus/get-active-connection-uuids.py	2020-08-19 09:39:11.000000000 +0200
+++ b/examples/python/dbus/get-active-connection-uuids.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2010 Red Hat, Inc.
diff -Naur a/examples/python/dbus/is-wwan-default.py b/examples/python/dbus/is-wwan-default.py
--- a/examples/python/dbus/is-wwan-default.py	2020-08-19 09:39:11.000000000 +0200
+++ b/examples/python/dbus/is-wwan-default.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2011 - 2012 Red Hat, Inc.
diff -Naur a/examples/python/dbus/list-connections.py b/examples/python/dbus/list-connections.py
--- a/examples/python/dbus/list-connections.py	2020-08-19 09:39:11.000000000 +0200
+++ b/examples/python/dbus/list-connections.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2010 - 2011 Red Hat, Inc.
diff -Naur a/examples/python/dbus/list-devices.py b/examples/python/dbus/list-devices.py
--- a/examples/python/dbus/list-devices.py	2020-08-19 09:39:11.000000000 +0200
+++ b/examples/python/dbus/list-devices.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2011 - 2012 Red Hat, Inc.
diff -Naur a/examples/python/dbus/nm-state.py b/examples/python/dbus/nm-state.py
--- a/examples/python/dbus/nm-state.py	2020-08-19 09:39:11.000000000 +0200
+++ b/examples/python/dbus/nm-state.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2010 - 2012 Red Hat, Inc.
diff -Naur a/examples/python/dbus/show-bssids.py b/examples/python/dbus/show-bssids.py
--- a/examples/python/dbus/show-bssids.py	2020-08-19 09:39:11.000000000 +0200
+++ b/examples/python/dbus/show-bssids.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2010 Red Hat, Inc.
diff -Naur a/examples/python/dbus/update-ip4-method.py b/examples/python/dbus/update-ip4-method.py
--- a/examples/python/dbus/update-ip4-method.py	2020-08-18 23:05:13.000000000 +0200
+++ b/examples/python/dbus/update-ip4-method.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2014 Red Hat, Inc.
diff -Naur a/examples/python/dbus/update-secrets.py b/examples/python/dbus/update-secrets.py
--- a/examples/python/dbus/update-secrets.py	2020-08-19 09:39:11.000000000 +0200
+++ b/examples/python/dbus/update-secrets.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2011 - 2012 Red Hat, Inc.
diff -Naur a/examples/python/dbus/vpn.py b/examples/python/dbus/vpn.py
--- a/examples/python/dbus/vpn.py	2020-08-19 09:39:11.000000000 +0200
+++ b/examples/python/dbus/vpn.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2009 Novell, Inc.
diff -Naur a/examples/python/dbus/wifi-active-ap.py b/examples/python/dbus/wifi-active-ap.py
--- a/examples/python/dbus/wifi-active-ap.py	2020-08-18 23:05:13.000000000 +0200
+++ b/examples/python/dbus/wifi-active-ap.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2015 Red Hat, Inc.
diff -Naur a/examples/python/dbus/wifi-hotspot.py b/examples/python/dbus/wifi-hotspot.py
--- a/examples/python/dbus/wifi-hotspot.py	2020-08-19 09:39:11.000000000 +0200
+++ b/examples/python/dbus/wifi-hotspot.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2010 - 2012 Red Hat, Inc.
diff -Naur a/examples/python/gi/add_connection.py b/examples/python/gi/add_connection.py
--- a/examples/python/gi/add_connection.py	2020-08-18 23:05:13.000000000 +0200
+++ b/examples/python/gi/add_connection.py	2020-09-28 14:46:04.393212851 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2014 Red Hat, Inc.
diff -Naur a/examples/python/gi/deactivate-all.py b/examples/python/gi/deactivate-all.py
--- a/examples/python/gi/deactivate-all.py	2020-08-18 23:05:13.000000000 +0200
+++ b/examples/python/gi/deactivate-all.py	2020-09-28 14:46:04.393212851 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2015 Red Hat, Inc.
diff -Naur a/examples/python/gi/device-state-ip4config.py b/examples/python/gi/device-state-ip4config.py
--- a/examples/python/gi/device-state-ip4config.py	2020-08-18 23:05:13.000000000 +0200
+++ b/examples/python/gi/device-state-ip4config.py	2020-09-28 14:46:04.393212851 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2012 Red Hat, Inc.
diff -Naur a/examples/python/gi/firewall-zone.py b/examples/python/gi/firewall-zone.py
--- a/examples/python/gi/firewall-zone.py	2020-08-18 23:05:13.000000000 +0200
+++ b/examples/python/gi/firewall-zone.py	2020-09-28 14:46:04.393212851 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2013 - 2014 Red Hat, Inc.
diff -Naur a/examples/python/gi/get-active-connections.py b/examples/python/gi/get-active-connections.py
--- a/examples/python/gi/get-active-connections.py	2020-08-18 23:05:13.000000000 +0200
+++ b/examples/python/gi/get-active-connections.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2014 Red Hat, Inc.
diff -Naur a/examples/python/gi/get_ips.py b/examples/python/gi/get_ips.py
--- a/examples/python/gi/get_ips.py	2020-08-18 23:05:13.000000000 +0200
+++ b/examples/python/gi/get_ips.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2014 Red Hat, Inc.
diff -Naur a/examples/python/gi/list-connections.py b/examples/python/gi/list-connections.py
--- a/examples/python/gi/list-connections.py	2020-08-18 23:05:13.000000000 +0200
+++ b/examples/python/gi/list-connections.py	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2012 - 2014 Red Hat, Inc.
diff -Naur a/examples/python/gi/nm-add-connection2.py b/examples/python/gi/nm-add-connection2.py
--- a/examples/python/gi/nm-add-connection2.py	2020-08-18 23:05:13.000000000 +0200
+++ b/examples/python/gi/nm-add-connection2.py	2020-09-28 14:46:04.393212851 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2019 Red Hat, Inc.
diff -Naur a/examples/python/gi/nm-connection-update-stable-id.py b/examples/python/gi/nm-connection-update-stable-id.py
--- a/examples/python/gi/nm-connection-update-stable-id.py	2020-08-18 23:05:13.000000000 +0200
+++ b/examples/python/gi/nm-connection-update-stable-id.py	2020-09-28 14:46:04.393212851 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2017 Red Hat, Inc.
diff -Naur a/examples/python/gi/nm-update2.py b/examples/python/gi/nm-update2.py
--- a/examples/python/gi/nm-update2.py	2020-08-18 23:05:13.000000000 +0200
+++ b/examples/python/gi/nm-update2.py	2020-09-28 14:46:04.393212851 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2019 Red Hat, Inc.
diff -Naur a/examples/python/gi/nm-wg-set b/examples/python/gi/nm-wg-set
--- a/examples/python/gi/nm-wg-set	2020-08-18 23:05:13.000000000 +0200
+++ b/examples/python/gi/nm-wg-set	2020-09-28 14:46:04.389879518 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2018 - 2019 Red Hat, Inc.
diff -Naur a/examples/python/gi/setting-user-data.py b/examples/python/gi/setting-user-data.py
--- a/examples/python/gi/setting-user-data.py	2020-08-18 23:05:13.000000000 +0200
+++ b/examples/python/gi/setting-user-data.py	2020-09-28 14:46:04.393212851 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2017 Red Hat, Inc.
diff -Naur a/examples/python/gi/show-wifi-networks.py b/examples/python/gi/show-wifi-networks.py
--- a/examples/python/gi/show-wifi-networks.py	2020-08-18 23:05:13.000000000 +0200
+++ b/examples/python/gi/show-wifi-networks.py	2020-09-28 14:46:04.393212851 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # coding=utf-8
 # SPDX-License-Identifier: GPL-2.0+
 #
diff -Naur a/examples/python/gi/update-ip4-method.py b/examples/python/gi/update-ip4-method.py
--- a/examples/python/gi/update-ip4-method.py	2020-08-18 23:05:13.000000000 +0200
+++ b/examples/python/gi/update-ip4-method.py	2020-09-28 14:46:04.393212851 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: GPL-2.0+
 #
 # Copyright (C) 2014 Red Hat, Inc.
diff -Naur a/src/meson.build b/src/meson.build
--- a/src/meson.build	2020-08-19 09:39:11.000000000 +0200
+++ b/src/meson.build	2020-09-28 14:45:50.919879595 +0200
@@ -251,7 +251,6 @@
 endif
 
 subdir('devices')
-subdir('initrd')
 subdir('settings/plugins')
 
 # NetworkManager binary
diff -Naur a/src/tests/test-secret-agent.py b/src/tests/test-secret-agent.py
--- a/src/tests/test-secret-agent.py	2020-08-18 23:05:14.000000000 +0200
+++ b/src/tests/test-secret-agent.py	2020-09-28 14:46:04.393212851 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 from gi.repository import GLib
 import sys
diff -Naur a/tools/debug-helper.py b/tools/debug-helper.py
--- a/tools/debug-helper.py	2020-08-18 23:05:14.000000000 +0200
+++ b/tools/debug-helper.py	2020-09-28 14:46:04.386546184 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # Copyright (C) 2011 Mathieu Trudel-Lapierre <mathieu.tl@gmail.com>
 
 import dbus
diff -Naur a/tools/generate-docs-nm-settings-docs-gir.py b/tools/generate-docs-nm-settings-docs-gir.py
--- a/tools/generate-docs-nm-settings-docs-gir.py	2020-08-18 23:05:14.000000000 +0200
+++ b/tools/generate-docs-nm-settings-docs-gir.py	2020-09-28 14:50:26.229878298 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: LGPL-2.1+
 #
 # Copyright (C) 2009 - 2017 Red Hat, Inc.
@@ -244,8 +244,8 @@
             "%s needs a gtk-doc block with one-line description" % setting.props.name
         )
     outfile.write(
-        '  <setting name="%s" description="%s" name_upper="%s" >\n'
-        % (setting.props.name, class_desc, get_setting_name_define(settingxml))
+        '  <setting name="%s" description=%s name_upper="%s" >\n'
+        % (setting.props.name, xml_quoteattr(class_desc), get_setting_name_define(settingxml))
     )
 
     setting_properties = {
diff -Naur a/tools/generate-docs-nm-settings-docs-merge.py b/tools/generate-docs-nm-settings-docs-merge.py
--- a/tools/generate-docs-nm-settings-docs-merge.py	2020-08-18 23:05:14.000000000 +0200
+++ b/tools/generate-docs-nm-settings-docs-merge.py	2020-09-28 14:46:04.386546184 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: LGPL-2.1+
 
 from __future__ import print_function
diff -Naur a/tools/test-networkmanager-service.py b/tools/test-networkmanager-service.py
--- a/tools/test-networkmanager-service.py	2020-08-18 23:05:14.000000000 +0200
+++ b/tools/test-networkmanager-service.py	2020-09-28 14:46:04.386546184 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 from __future__ import print_function
 
