diff -Naur a/Telegram/ThirdParty/tgcalls/tgcalls/utils/gzip.h b/Telegram/ThirdParty/tgcalls/tgcalls/utils/gzip.h
--- a/Telegram/ThirdParty/tgcalls/tgcalls/utils/gzip.h	2022-06-20 18:24:08.000000000 +0200
+++ b/Telegram/ThirdParty/tgcalls/tgcalls/utils/gzip.h	2022-06-28 13:04:24.343374836 +0200
@@ -3,6 +3,7 @@
 
 #include <absl/types/optional.h>
 #include <vector>
+#include <cstdint>
 
 namespace tgcalls {
 
