--- a/src/efibootmgr.c	2018-06-10 20:12:10.000000000 +0000
+++ b/src/efibootmgr.c	2020-01-01 14:13:46.073302977 +0000
@@ -1536,10 +1536,7 @@
 					       "invalid numeric value %s\n",
 					       optarg);
 			}
-                        /* XXX efivar-36 accidentally doesn't have a public
-                         * header for this */
-			extern int efi_set_verbose(int verbosity, FILE *errlog);
-			efi_set_verbose(opts.verbose - 2, stderr);
+			efi_set_verbose(opts.verbose - 1, stderr);
 			break;
 		case 'V':
 			opts.showversion = 1;
