diff -Naur qemu-2.11.1-original/configure qemu-2.11.1/configure
--- qemu-2.11.1-original/configure	2018-02-14 21:53:22.000000000 +0000
+++ qemu-2.11.1/configure	2018-02-20 21:25:14.159990856 +0000
@@ -3920,7 +3920,7 @@
 # check if memfd is supported
 memfd=no
 cat > $TMPC << EOF
-#include <sys/memfd.h>
+#include <sys/mman.h>
 
 int main(void)
 {
diff -Naur qemu-2.11.1-original/util/memfd.c qemu-2.11.1/util/memfd.c
--- qemu-2.11.1-original/util/memfd.c	2018-02-14 21:53:23.000000000 +0000
+++ qemu-2.11.1/util/memfd.c	2018-02-20 21:26:47.647070550 +0000
@@ -31,9 +31,7 @@
 
 #include "qemu/memfd.h"
 
-#ifdef CONFIG_MEMFD
-#include <sys/memfd.h>
-#elif defined CONFIG_LINUX
+#if defined CONFIG_LINUX && !defined CONFIG_MEMFD
 #include <sys/syscall.h>
 #include <asm/unistd.h>
 
