diff -Naur blfs-bootscripts-20150304.original/blfs/init.d/sshd blfs-bootscripts-20150304/blfs/init.d/sshd
--- blfs-bootscripts-20150304.original/blfs/init.d/sshd	2012-06-18 23:42:28.000000000 +0000
+++ blfs-bootscripts-20150304/blfs/init.d/sshd	2015-03-14 20:09:10.016403906 +0000
@@ -30,7 +30,20 @@
 
 case "$1" in
     start)
+
         log_info_msg "Starting SSH Server..."
+	if [ ! -f /etc/ssh/ssh_host_rsa_key ]; then
+        	/usr/bin/ssh-keygen -t rsa -N "" -f /etc/ssh/ssh_host_rsa_key
+	fi
+	if [ ! -f /etc/ssh/ssh_host_dsa_key ]; then
+        	/usr/bin/ssh-keygen -t dsa -N "" -f /etc/ssh/ssh_host_dsa_key
+	fi
+	if [ ! -f /etc/ssh/ssh_host_ecdsa_key ]; then
+        	/usr/bin/ssh-keygen -t ecdsa -N "" -f /etc/ssh/ssh_host_ecdsa_key
+	fi
+	if [ ! -f /etc/ssh/ssh_host_ed25519_key ]; then
+        	/usr/bin/ssh-keygen -t ed25519 -N "" -f /etc/ssh/ssh_host_ed25519_key
+	fi
         start_daemon -f /usr/sbin/sshd
         evaluate_retval
         # Also prevent ssh from being killed by out of memory conditions
