diff -Naur kholidays-15.12.3-orig/src/sunriseset.cpp kholidays-15.12.3/src/sunriseset.cpp
--- kholidays-15.12.3-orig/src/sunriseset.cpp	2016-02-11 18:26:09.000000000 +0100
+++ kholidays-15.12.3/src/sunriseset.cpp	2016-03-20 21:48:38.699876214 +0100
@@ -21,6 +21,7 @@
   the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
   Boston, MA 02110-1301, USA.
 */
+using namespace std;
 
 #include "sunriseset.h"
 #include <math.h>
