diff -Naur mate-desktop-1.18.0-orig/schemas/org.mate.background.gschema.xml.in mate-desktop-1.18.0/schemas/org.mate.background.gschema.xml.in
--- mate-desktop-1.18.0-orig/schemas/org.mate.background.gschema.xml.in	2017-03-13 09:05:01.000000000 +0000
+++ mate-desktop-1.18.0/schemas/org.mate.background.gschema.xml.in	2017-08-28 19:52:35.303278000 +0000
@@ -29,12 +29,12 @@
       <description>If set to true, then MATE will change the desktop background with a fading effect.</description>
     </key>
     <key name="picture-options" enum="org.mate.background.picture-options-enum">
-      <default>'zoom'</default>
+      <default>'scaled'</default>
       <summary>Picture Options</summary>
       <description>Determines how the image set by wallpaper_filename is rendered.  Possible values are "wallpaper", "centered", "scaled", "stretched", "zoom", "spanned".</description>
     </key>
     <key name="picture-filename" type="s">
-      <default>'@datadir@/backgrounds/mate/desktop/Stripes.png'</default>
+      <default>'/usr/share/backgrounds/mate/desktop/mate-nutyx.png'</default>
       <summary>Picture Filename</summary>
       <description>File to use for the background image.</description>
     </key>
