diff -Naur slim-1.3.6-original/CMakeLists.txt slim-1.3.6/CMakeLists.txt
--- slim-1.3.6-original/CMakeLists.txt	2013-10-02 00:38:05.000000000 +0200
+++ slim-1.3.6/CMakeLists.txt	2018-01-12 22:38:24.266653668 +0100
@@ -119,6 +119,7 @@
 	if(PAM_FOUND)
 		message("\tPAM Found")
 		set(SLIM_DEFINITIONS ${SLIM_DEFINITIONS} "-DUSE_PAM")
+		target_link_libraries(libslim ${PAM_LIBRARY})
 		target_link_libraries(${PROJECT_NAME} ${PAM_LIBRARY})
 		target_link_libraries(slimlock ${PAM_LIBRARY})
 		include_directories(${PAM_INCLUDE_DIR})
@@ -178,6 +179,8 @@
 )
 
 target_link_libraries(libslim
+	${X11_Xft_LIB}
+	${X11_Xrandr_LIB}
     ${JPEG_LIBRARIES}
 	${PNG_LIBRARIES}
 )
diff -Naur slim-1.3.6-original/slim.conf slim-1.3.6/slim.conf
--- slim-1.3.6-original/slim.conf	2013-10-02 00:38:05.000000000 +0200
+++ slim-1.3.6/slim.conf	2018-01-12 22:38:32.083320295 +0100
@@ -47,11 +47,9 @@
 # options "-d" and "-nodaemon"
 # daemon	yes
 
-# Available sessions (first one is the default).
-# The current chosen session name is replaced in the login_cmd
-# above, so your login command can handle different sessions.
-# see the xinitrc.sample file shipped with slim sources
-sessions            xfce4,icewm-session,wmaker,blackbox
+# Set directory that contains the xsessions.
+# slim reads xsesion from this directory, and be able to select.
+sessiondir            /usr/share/xsessions/
 
 # Executed when pressing F11 (requires imagemagick)
 screenshot_cmd      import -window root /slim.png
diff -Naur slim-1.3.6-original/themes/default/slim.theme slim-1.3.6/themes/default/slim.theme
--- slim-1.3.6-original/themes/default/slim.theme	2013-10-02 00:38:05.000000000 +0200
+++ slim-1.3.6/themes/default/slim.theme	2018-01-12 22:40:43.063319693 +0100
@@ -4,8 +4,8 @@
 # Messages (ie: shutdown)
 msg_color               #FFFFFF
 msg_font                Verdana:size=18:bold:dpi=75
-msg_x                   50%
-msg_y                   40%
+msg_x                   25%
+msg_y                   50%
 msg_shadow_color #702342
 msg_shadow_xoffset 1
 msg_shadow_yoffset 1
@@ -15,23 +15,23 @@
 background_color        #eedddd
 
 # Input controls
-input_panel_x           25%
-input_panel_y           65%
-input_name_x            394
-input_name_y            181
-input_font          	Verdana:size=12:dpi=75
-input_color             #000000
+input_panel_x           50%
+input_panel_y           50%
+input_name_x            125
+input_name_y            252
+input_font          	Verdana:size=21:bold:dpi=75
+input_color             #FFFFFF
 
 # Username / password request
-username_font          	Verdana:size=14:bold:dpi=75
-username_color        	#f9f9f9
-username_x              280
-username_y              183
-password_x              50%
-password_y              183
-username_shadow_color   #702342
-username_shadow_xoffset 1
-username_shadow_yoffset 1
+username_font          	Verdana:size=21:bold:dpi=75
+username_color        	#FFFFFF
+username_x              0%
+username_y              100%
+password_x              0%
+password_y              100%
+username_shadow_color   #000000
+username_shadow_xoffset 2
+username_shadow_yoffset 3
 
 username_msg            Username:
 password_msg            Password:
