--- d/startlxqt.in	2022-04-16 15:16:20.000000000 +0200
+++ b/startlxqt.in	2022-04-28 18:06:40.548967287 +0200
@@ -29,6 +29,9 @@
     if ! contains "$XDG_CONFIG_DIRS" '@LXQT_ETC_XDG_DIR@'; then
         XDG_CONFIG_DIRS="$XDG_CONFIG_DIRS:@LXQT_ETC_XDG_DIR@"
     fi
+    if ! contains "$XDG_CONFIG_DIRS" '@LXQT_DATA_DIR@'; then
+        XDG_CONFIG_DIRS="$XDG_CONFIG_DIRS:@LXQT_DATA_DIR@"
+    fi
 fi
 
 if [ -z "$XDG_CACHE_HOME" ]; then
