diff -Naur a/startlxqt.in b/startlxqt.in
--- a/startlxqt.in	2021-04-16 18:36:14.000000000 +0200
+++ b/startlxqt.in	2021-09-20 14:01:25.779885740 +0200
@@ -29,6 +29,9 @@
     if ! contains "$XDG_CONFIG_DIRS" '@LXQT_ETC_XDG_DIR@'; then
         XDG_CONFIG_DIRS="$XDG_CONFIG_DIRS:@LXQT_ETC_XDG_DIR@"
     fi
+    if ! contains "$XDG_CONFIG_DIRS" '@LXQT_DATA_DIR@'; then
+        XDG_CONFIG_DIRS="$XDG_CONFIG_DIRS:@LXQT_DATA_DIR@"
+    fi
 fi
 
 if [ -z "$XDG_CACHE_HOME" ]; then
