diff -Naur a/hw/xquartz/X11Controller.m.rej b/hw/xquartz/X11Controller.m.rej
--- a/hw/xquartz/X11Controller.m.rej	1970-01-01 01:00:00.000000000 +0100
+++ b/hw/xquartz/X11Controller.m.rej	2022-11-01 13:00:38.229225892 +0100
@@ -0,0 +1,16 @@
+--- hw/xquartz/X11Controller.m
++++ hw/xquartz/X11Controller.m
+@@ -469,8 +469,11 @@ extern char *bundle_id_prefix;
+     self.table_apps = table_apps;
+ 
+     NSArray * const apps = self.apps;
+-    if (apps != nil)
+-        [table_apps addObjectsFromArray:apps];
++    if (apps != nil) {
++        for (NSArray <NSString *> * row in apps) {
++            [table_apps addObject:row.mutableCopy];
++        }
++    }
+ 
+     columns = [apps_table tableColumns];
+     [[columns objectAtIndex:0] setIdentifier:@"0"];
