diff -rupN a/data/etc/e-applications.menu b/data/etc/e-applications.menu
--- a/data/etc/e-applications.menu	2015-09-28 22:11:17.000000000 +0200
+++ b/data/etc/e-applications.menu	2015-11-19 12:21:04.709461076 +0100
@@ -20,7 +20,7 @@
 
   <!-- Accessories submenu -->
   <Menu>
-    <Name>Accessories</Name>
+    <Name>Accessoires</Name>
     <Directory>Utility.directory</Directory>
     <Include>
       <And>
@@ -38,7 +38,7 @@
 
   <!-- Accessibility submenu -->
   <Menu>
-    <Name>Universal Access</Name>
+    <Name>Accessibilité</Name>
     <Directory>Utility-Accessibility.directory</Directory>
     <Include>
       <And>
@@ -50,7 +50,7 @@
 
   <!-- Development Tools -->
   <Menu>
-    <Name>Development</Name>
+    <Name>Développement</Name>
     <Directory>Development.directory</Directory>
     <Include>
       <Or>
@@ -97,7 +97,7 @@
 
   <!-- Games -->
   <Menu>
-    <Name>Games</Name>
+    <Name>Jeux</Name>
     <Directory>Game.directory</Directory>
     <Include>
       <And>
@@ -130,7 +130,7 @@
       </Include>
     </Menu>
     <Menu>
-      <Name>Adventure</Name>
+      <Name>Aventure</Name>
       <Directory>AdventureGames.directory</Directory>
       <Include>
         <Category>AdventureGame</Category>
@@ -144,7 +144,7 @@
       </Include>
     </Menu>
     <Menu>
-      <Name>Board</Name>
+      <Name>Plateau</Name>
       <Directory>BoardGames.directory</Directory>
       <Include>
         <Category>BoardGame</Category>
@@ -158,14 +158,14 @@
       </Include>
     </Menu>
     <Menu>
-      <Name>Cards</Name>
+      <Name>Cartes</Name>
       <Directory>CardGames.directory</Directory>
       <Include>
         <Category>CardGame</Category>
       </Include>
     </Menu>
     <Menu>
-      <Name>Kids</Name>
+      <Name>Enfants</Name>
       <Directory>KidsGames.directory</Directory>
       <Include>
         <Category>KidsGame</Category>
@@ -173,14 +173,14 @@
       </Include>
     </Menu>
     <Menu>
-      <Name>Logic</Name>
+      <Name>Logique</Name>
       <Directory>LogicGames.directory</Directory>
       <Include>
         <Category>LogicGame</Category>
       </Include>
     </Menu>
     <Menu>
-      <Name>Role Playing</Name>
+      <Name>Role</Name>
       <Directory>RolePlayingGames.directory</Directory>
       <Include>
         <Category>RolePlaying</Category>
@@ -201,7 +201,7 @@
       </Include>
     </Menu>
     <Menu>
-      <Name>Strategy</Name>
+      <Name>Stratégie</Name>
       <Directory>StrategyGames.directory</Directory>
       <Include>
         <Category>StrategyGame</Category>
@@ -211,7 +211,7 @@
 
   <!-- Graphics -->
   <Menu>
-    <Name>Graphics</Name>
+    <Name>Graphisme</Name>
     <Directory>Graphics.directory</Directory>
     <Include>
       <And>
@@ -233,7 +233,7 @@
 
   <!-- Multimedia -->
   <Menu>
-    <Name>Multimedia</Name>
+    <Name>Multimédia</Name>
     <Directory>AudioVideo.directory</Directory>
     <Include>
       <And>
@@ -244,7 +244,7 @@
 
   <!-- Office -->
   <Menu>
-    <Name>Office</Name>
+    <Name>Bureautique</Name>
     <Directory>Office.directory</Directory>
     <Include>
       <And>
@@ -255,7 +255,7 @@
 
   <!-- System Tools-->
   <Menu>
-    <Name>System</Name>
+    <Name>Système</Name>
     <Directory>System-Tools.directory</Directory>
     <Include>
       <And>
@@ -268,7 +268,7 @@
 
   <!-- Other -->
   <Menu>
-    <Name>Other</Name>
+    <Name>Autres</Name>
     <Directory>Other.directory</Directory>
     <OnlyUnallocated/>
     <Include>
diff -rupN a/src/modules/wizard/page_060.c b/src/modules/wizard/page_060.c
--- a/src/modules/wizard/page_060.c	2015-09-28 22:11:18.000000000 +0200
+++ b/src/modules/wizard/page_060.c	2015-11-19 18:08:49.558943220 +0100
@@ -1,7 +1,7 @@
 /* Ask about focus mode */
 #include "e_wizard.h"
 
-static int focus_mode = 1;
+static int focus_mode = 0;
 /*
 E_API int
 wizard_page_init(E_Wizard_Page *pg __UNUSED__, Eina_Bool *need_xdg_desktops __UNUSED__, Eina_Bool *need_xdg_icons __UNUSED__)
diff -rupN a/src/modules/wizard/page_110.c b/src/modules/wizard/page_110.c
--- a/src/modules/wizard/page_110.c	2015-09-28 22:11:18.000000000 +0200
+++ b/src/modules/wizard/page_110.c	2015-11-19 12:27:11.576118637 +0100
@@ -158,10 +158,10 @@ wizard_page_show(E_Wizard_Page *pg)
                }
           }
         e_config_save_queue();
-        _recommend_connman(pg);
+      //  _recommend_connman(pg);
      }
    e_wizard_title_set(_("Checking to see if Connman exists"));
-   return 1; /* 1 == show ui, and wait for user, 0 == just continue */
+   return 0; /* 1 == show ui, and wait for user, 0 == just continue */
 }
 
 E_API int
