diff -Naur glib-2.54.0-orig/gio/glib-compile-schemas.c glib-2.54.0/gio/glib-compile-schemas.c
--- glib-2.54.0-orig/gio/glib-compile-schemas.c	2017-08-07 14:58:31.000000000 +0000
+++ glib-2.54.0/gio/glib-compile-schemas.c	2017-09-18 09:36:02.194079754 +0000
@@ -1219,19 +1219,6 @@
       return;
     }
 
-  if (path && (g_str_has_prefix (path, "/apps/") ||
-               g_str_has_prefix (path, "/desktop/") ||
-               g_str_has_prefix (path, "/system/")))
-    {
-      gchar *message = NULL;
-      message = g_strdup_printf (_("Warning: Schema “%s” has path “%s”.  "
-                                   "Paths starting with "
-                                   "“/apps/”, “/desktop/” or “/system/” are deprecated."),
-                                 id, path);
-      g_printerr ("%s\n", message);
-      g_free (message);
-    }
-
   state->schema_state = schema_state_new (path, gettext_domain,
                                           extends, extends_name, list_of);
 
