--- a/CMakeLists.txt	2015-11-14 11:18:41.030039651 -0800
+++ b/CMakeLists.txt	2015-11-14 11:18:20.410039499 -0800
@@ -110,9 +110,9 @@
 #FINDAVUTIL("avutil")
 
 add_definitions( -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS)
-add_subdirectory( src/libdvdnav-NOW) #from dvdnav
-add_subdirectory( src/libdvdnav-NOW/dvdnav)
-add_subdirectory( src/libdvdnav-NOW/vm)
+#add_subdirectory( src/libdvdnav-NOW) #from dvdnav
+#add_subdirectory( src/libdvdnav-NOW/dvdnav)
+#add_subdirectory( src/libdvdnav-NOW/vm)
 add_subdirectory( src/core )
 add_subdirectory( src/mpeg2)
 add_subdirectory(src/vamps)
