diff --git a/fltk/table_collection.cxx b/fltk/table_collection.cxx
index 1660cae..b27f6a9 100644
--- a/fltk/table_collection.cxx
+++ b/fltk/table_collection.cxx
@@ -175,7 +175,7 @@ void TableCollection::OnEvent(TableContext context, int pCol, int pRow)
                     Collec_List.insert("xfce4");
                     Collec_List.insert("xfce4-extra");
                }
-                else if (Collection=="E17")
+                else if (Collection=="E24")
                 {
                     Collec_List.insert("enlightenment");
                 }
